// @alfresco/adf-core/esm2015/form/components/widgets/amount/amount.widget.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ViewEncapsulation, InjectionToken, Inject, Optional } from '@angular/core';
import { FormService } from './../../../services/form.service';
import { WidgetComponent } from './../widget.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './../../../services/form.service';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '@angular/material/form-field';
import * as ɵngcc4 from '@angular/material/input';
import * as ɵngcc5 from '@angular/forms';
import * as ɵngcc6 from '@angular/material/tooltip';
import * as ɵngcc7 from '../error/error.component';
import * as ɵngcc8 from '@ngx-translate/core';

function AmountWidgetComponent_span_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1, "*");
    ɵngcc0.ɵɵelementEnd();
} }
function AmountWidgetComponent_error_widget_10_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "error-widget", 7);
    ɵngcc0.ɵɵpipe(1, "translate");
} if (rf & 2) {
    ɵngcc0.ɵɵpropertyInterpolate("required", ɵngcc0.ɵɵpipeBind1(1, 1, "FORM.FIELD.REQUIRED"));
} }
export const ADF_AMOUNT_SETTINGS = new InjectionToken('adf-amount-settings');
export class AmountWidgetComponent extends WidgetComponent {
    constructor(formService, settings) {
        super(formService);
        this.formService = formService;
        this.settings = settings;
        this.showPlaceholder = true;
        this.currency = AmountWidgetComponent.DEFAULT_CURRENCY;
    }
    get placeholder() {
        return this.showPlaceholder ? this.field.placeholder : '';
    }
    ngOnInit() {
        if (this.field) {
            if (this.field.currency) {
                this.currency = this.field.currency;
            }
            if (this.field.readOnly) {
                this.showPlaceholder = this.settings && this.settings.showReadonlyPlaceholder;
            }
        }
    }
}
AmountWidgetComponent.ɵfac = function AmountWidgetComponent_Factory(t) { return new (t || AmountWidgetComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormService), ɵngcc0.ɵɵdirectiveInject(ADF_AMOUNT_SETTINGS, 8)); };
AmountWidgetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AmountWidgetComponent, selectors: [["amount-widget"]], hostBindings: function AmountWidgetComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function AmountWidgetComponent_click_HostBindingHandler($event) { return ctx.event($event); })("blur", function AmountWidgetComponent_blur_HostBindingHandler($event) { return ctx.event($event); })("change", function AmountWidgetComponent_change_HostBindingHandler($event) { return ctx.event($event); })("focus", function AmountWidgetComponent_focus_HostBindingHandler($event) { return ctx.event($event); })("focusin", function AmountWidgetComponent_focusin_HostBindingHandler($event) { return ctx.event($event); })("focusout", function AmountWidgetComponent_focusout_HostBindingHandler($event) { return ctx.event($event); })("input", function AmountWidgetComponent_input_HostBindingHandler($event) { return ctx.event($event); })("invalid", function AmountWidgetComponent_invalid_HostBindingHandler($event) { return ctx.event($event); })("select", function AmountWidgetComponent_select_HostBindingHandler($event) { return ctx.event($event); });
    } }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 11, vars: 22, consts: [[1, "adf-label"], [4, "ngIf"], [1, "adf-amount-widget__input"], ["matPrefix", "", 1, "adf-amount-widget__prefix-spacing"], ["matInput", "", "matTooltipPosition", "above", "matTooltipShowDelay", "1000", "type", "text", 1, "adf-input", 3, "matTooltip", "id", "required", "placeholder", "value", "ngModel", "disabled", "ngModelChange"], [3, "error"], [3, "required", 4, "ngIf"], [3, "required"]], template: function AmountWidgetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵelementStart(1, "label", 0);
        ɵngcc0.ɵɵtext(2);
        ɵngcc0.ɵɵpipe(3, "translate");
        ɵngcc0.ɵɵtemplate(4, AmountWidgetComponent_span_4_Template, 2, 0, "span", 1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "mat-form-field", 2);
        ɵngcc0.ɵɵelementStart(6, "span", 3);
        ɵngcc0.ɵɵtext(7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(8, "input", 4);
        ɵngcc0.ɵɵlistener("ngModelChange", function AmountWidgetComponent_Template_input_ngModelChange_8_listener($event) { return ctx.field.value = $event; })("ngModelChange", function AmountWidgetComponent_Template_input_ngModelChange_8_listener() { return ctx.onFieldChanged(ctx.field); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(9, "error-widget", 5);
        ɵngcc0.ɵɵtemplate(10, AmountWidgetComponent_error_widget_10_Template, 2, 3, "error-widget", 6);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMapInterpolate1("adf-amount-widget__container adf-amount-widget ", ctx.field.className, "");
        ɵngcc0.ɵɵclassProp("adf-invalid", !ctx.field.isValid)("adf-readonly", ctx.field.readOnly);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵattribute("for", ctx.field.id);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(3, 20, ctx.field.name));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isRequired());
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1("", ctx.currency, " \u00A0");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("matTooltip", ctx.field.tooltip)("id", ctx.field.id)("required", ctx.isRequired())("placeholder", ctx.placeholder)("value", ctx.field.value)("ngModel", ctx.field.value)("disabled", ctx.field.readOnly);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("error", ctx.field.validationSummary);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isInvalidFieldRequired());
    } }, directives: [ɵngcc2.NgIf, ɵngcc3.MatFormField, ɵngcc3.MatPrefix, ɵngcc4.MatInput, ɵngcc5.DefaultValueAccessor, ɵngcc6.MatTooltip, ɵngcc5.RequiredValidator, ɵngcc5.NgControlStatus, ɵngcc5.NgModel, ɵngcc7.ErrorWidgetComponent], pipes: [ɵngcc8.TranslatePipe], encapsulation: 2 });
AmountWidgetComponent.DEFAULT_CURRENCY = '$';
AmountWidgetComponent.ctorParameters = () => [
    { type: FormService },
    { type: undefined, decorators: [{ type: Inject, args: [ADF_AMOUNT_SETTINGS,] }, { type: Optional }] }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AmountWidgetComponent, [{
        type: Component,
        args: [{
                selector: 'amount-widget',
                template: "<div class=\"adf-amount-widget__container adf-amount-widget {{field.className}}\"\n     [class.adf-invalid]=\"!field.isValid\"\n     [class.adf-readonly]=\"field.readOnly\">\n    <label class=\"adf-label\"\n           [attr.for]=\"field.id\">{{field.name | translate }}<span *ngIf=\"isRequired()\">*</span></label>\n    <mat-form-field class=\"adf-amount-widget__input\">\n        <span matPrefix class=\"adf-amount-widget__prefix-spacing\">{{ currency }} &nbsp;</span>\n        <input matInput\n               [matTooltip]=\"field.tooltip\"\n               matTooltipPosition=\"above\"\n               matTooltipShowDelay=\"1000\"\n               class=\"adf-input\"\n               type=\"text\"\n               [id]=\"field.id\"\n               [required]=\"isRequired()\"\n               [placeholder]=\"placeholder\"\n               [value]=\"field.value\"\n               [(ngModel)]=\"field.value\"\n               (ngModelChange)=\"onFieldChanged(field)\"\n               [disabled]=\"field.readOnly\">\n    </mat-form-field>\n    <error-widget [error]=\"field.validationSummary\"></error-widget>\n    <error-widget *ngIf=\"isInvalidFieldRequired()\"\n                  required=\"{{ 'FORM.FIELD.REQUIRED' | translate }}\"></error-widget>\n</div>\n",
                host: {
                    '(click)': 'event($event)',
                    '(blur)': 'event($event)',
                    '(change)': 'event($event)',
                    '(focus)': 'event($event)',
                    '(focusin)': 'event($event)',
                    '(focusout)': 'event($event)',
                    '(input)': 'event($event)',
                    '(invalid)': 'event($event)',
                    '(select)': 'event($event)'
                },
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return [{ type: ɵngcc1.FormService }, { type: undefined, decorators: [{
                type: Inject,
                args: [ADF_AMOUNT_SETTINGS]
            }, {
                type: Optional
            }] }]; }, null); })();
