// @alfresco/adf-core/esm2015/templates/empty-content/empty-content.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, ChangeDetectionStrategy, ViewEncapsulation, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../icon/icon.component';
import * as ɵngcc2 from '@ngx-translate/core';

const _c0 = ["*"];
export class EmptyContentComponent {
    constructor() {
        this.icon = 'cake';
        this.title = '';
        this.subtitle = '';
    }
}
EmptyContentComponent.ɵfac = function EmptyContentComponent_Factory(t) { return new (t || EmptyContentComponent)(); };
EmptyContentComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: EmptyContentComponent, selectors: [["adf-empty-content"]], hostAttrs: [1, "adf-empty-content"], inputs: { icon: "icon", title: "title", subtitle: "subtitle" }, ngContentSelectors: _c0, decls: 9, vars: 7, consts: [[1, "adf-empty-content"], [1, "adf-empty-content__icon", 3, "value"], [1, "adf-empty-content__title"], [1, "adf-empty-content__subtitle"]], template: function EmptyContentComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "adf-icon", 1);
        ɵngcc0.ɵɵelementStart(2, "div", 2);
        ɵngcc0.ɵɵtext(3);
        ɵngcc0.ɵɵpipe(4, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 3);
        ɵngcc0.ɵɵtext(6);
        ɵngcc0.ɵɵpipe(7, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵprojection(8);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("value", ctx.icon);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(4, 3, ctx.title));
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(7, 5, ctx.subtitle));
    } }, directives: [ɵngcc1.IconComponent], pipes: [ɵngcc2.TranslatePipe], encapsulation: 2, changeDetection: 0 });
EmptyContentComponent.propDecorators = {
    icon: [{ type: Input }],
    title: [{ type: Input }],
    subtitle: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EmptyContentComponent, [{
        type: Component,
        args: [{
                selector: 'adf-empty-content',
                template: "<div class=\"adf-empty-content\">\n    <adf-icon class=\"adf-empty-content__icon\" [value]=\"icon\"></adf-icon>\n    <div class=\"adf-empty-content__title\">{{ title | translate }}</div>\n    <div class=\"adf-empty-content__subtitle\">{{ subtitle | translate }}</div>\n    <ng-content></ng-content>\n</div>\n",
                changeDetection: ChangeDetectionStrategy.OnPush,
                encapsulation: ViewEncapsulation.None,
                host: { class: 'adf-empty-content' }
            }]
    }], function () { return []; }, { icon: [{
            type: Input
        }], title: [{
            type: Input
        }], subtitle: [{
            type: Input
        }] }); })();
