// @alfresco/adf-content-services/esm2015/lib/permission-manager/components/add-permission/add-permission.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { AllowableOperationsEnum, ContentService } from '@alfresco/adf-core';
import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
import { NodePermissionService } from '../../services/node-permission.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../services/node-permission.service';
import * as ɵngcc2 from '@alfresco/adf-core';
import * as ɵngcc3 from './add-permission-panel.component';
import * as ɵngcc4 from '@angular/material/button';
import * as ɵngcc5 from '@ngx-translate/core';
export class AddPermissionComponent {
    constructor(nodePermissionService, contentService) {
        this.nodePermissionService = nodePermissionService;
        this.contentService = contentService;
        this.success = new EventEmitter();
        this.error = new EventEmitter();
        this.selectedItems = [];
    }
    ngOnInit() {
        this.nodePermissionService.getNodeWithRoles(this.nodeId).subscribe(({ node, roles }) => {
            this.currentNode = node;
            this.currentNodeRoles = roles;
        });
    }
    onSelect(selection) {
        this.selectedItems = selection;
    }
    isAddEnabled() {
        return this.contentService.hasAllowableOperations(this.currentNode, AllowableOperationsEnum.UPDATEPERMISSIONS) &&
            this.selectedItems.length !== 0;
    }
    applySelection() {
        if (this.contentService.hasAllowableOperations(this.currentNode, AllowableOperationsEnum.UPDATEPERMISSIONS)) {
            const permissions = this.transformNodeToPermissionElement(this.selectedItems, this.currentNodeRoles[0].role);
            this.nodePermissionService.updateNodePermissions(this.nodeId, permissions)
                .subscribe((node) => {
                this.success.emit(node);
            }, (error) => {
                this.error.emit(error);
            });
        }
    }
    transformNodeToPermissionElement(nodes, role) {
        return nodes.map((node) => {
            var _a;
            return {
                'authorityId': (_a = node.entry.properties['cm:authorityName']) !== null && _a !== void 0 ? _a : node.entry.properties['cm:userName'],
                'name': role,
                'accessStatus': 'ALLOWED'
            };
        });
    }
}
AddPermissionComponent.ɵfac = function AddPermissionComponent_Factory(t) { return new (t || AddPermissionComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NodePermissionService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.ContentService)); };
AddPermissionComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AddPermissionComponent, selectors: [["adf-add-permission"]], inputs: { nodeId: "nodeId" }, outputs: { success: "success", error: "error" }, decls: 5, vars: 4, consts: [[3, "select"], ["id", "adf-add-permission-actions"], ["mat-button", "", "id", "adf-add-permission-action-button", 1, "adf-permission-action", 3, "disabled", "click"]], template: function AddPermissionComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "adf-add-permission-panel", 0);
        ɵngcc0.ɵɵlistener("select", function AddPermissionComponent_Template_adf_add_permission_panel_select_0_listener($event) { return ctx.onSelect($event); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelementStart(2, "button", 2);
        ɵngcc0.ɵɵlistener("click", function AddPermissionComponent_Template_button_click_2_listener() { return ctx.applySelection(); });
        ɵngcc0.ɵɵtext(3);
        ɵngcc0.ɵɵpipe(4, "translate");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("disabled", !ctx.isAddEnabled());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(4, 2, "PERMISSION_MANAGER.ADD-PERMISSION.ADD-ACTION"), " ");
    } }, directives: [ɵngcc3.AddPermissionPanelComponent, ɵngcc4.MatButton], pipes: [ɵngcc5.TranslatePipe], encapsulation: 2 });
AddPermissionComponent.ctorParameters = () => [
    { type: NodePermissionService },
    { type: ContentService }
];
AddPermissionComponent.propDecorators = {
    nodeId: [{ type: Input }],
    success: [{ type: Output }],
    error: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AddPermissionComponent, [{
        type: Component,
        args: [{
                selector: 'adf-add-permission',
                template: "<adf-add-permission-panel\n    (select)=\"onSelect($event)\">\n</adf-add-permission-panel>\n<div id=\"adf-add-permission-actions\">\n   <button mat-button\n          id=\"adf-add-permission-action-button\"\n          class=\"adf-permission-action\"\n          [disabled]=\"!isAddEnabled()\"\n          (click)=\"applySelection()\">\n     {{'PERMISSION_MANAGER.ADD-PERMISSION.ADD-ACTION' | translate}}\n    </button>\n</div>\n",
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return [{ type: ɵngcc1.NodePermissionService }, { type: ɵngcc2.ContentService }]; }, { success: [{
            type: Output
        }], error: [{
            type: Output
        }], nodeId: [{
            type: Input
        }] }); })();
