// @alfresco/adf-core/esm2015/card-view/components/card-view-textitem/card-view-textitem.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ChangeDetectorRef, Component, Input } from '@angular/core';
import { CardViewUpdateService } from '../../services/card-view-update.service';
import { BaseCardView } from '../base-card-view';
import { ClipboardService } from '../../../clipboard/clipboard.service';
import { TranslationService } from '../../../services/translation.service';
import { FormControl } from '@angular/forms';
import { debounceTime, takeUntil, filter } from 'rxjs/operators';
import { Subject } from 'rxjs';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../services/card-view-update.service';
import * as ɵngcc2 from '../../../clipboard/clipboard.service';
import * as ɵngcc3 from '../../../services/translation.service';
import * as ɵngcc4 from '@angular/common';
import * as ɵngcc5 from '@angular/material/form-field';
import * as ɵngcc6 from '@angular/flex-layout/extended';
import * as ɵngcc7 from '@angular/material/input';
import * as ɵngcc8 from '@angular/forms';
import * as ɵngcc9 from '@angular/material/tooltip';
import * as ɵngcc10 from '@angular/material/button';
import * as ɵngcc11 from '@angular/material/icon';
import * as ɵngcc12 from '@angular/material/chips';
import * as ɵngcc13 from '@angular/flex-layout/flex';
import * as ɵngcc14 from '@ngx-translate/core';

function CardViewTextItemComponent_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 7);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵattribute("data-automation-id", "card-textitem-label-" + ctx_r0.property.key);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 2, ctx_r0.property.label), " ");
} }
function CardViewTextItemComponent_div_2_input_2_Template(rf, ctx) { if (rf & 1) {
    const _r11 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 13);
    ɵngcc0.ɵɵlistener("dblclick", function CardViewTextItemComponent_div_2_input_2_Template_input_dblclick_0_listener() { ɵngcc0.ɵɵrestoreView(_r11); const ctx_r10 = ɵngcc0.ɵɵnextContext(2); return ctx_r10.copyToClipboard(ctx_r10.property.displayValue); });
    ɵngcc0.ɵɵpipe(1, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r6 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("placeholder", ctx_r6.property.default)("formControl", ctx_r6.textInput)("matTooltip", ɵngcc0.ɵɵpipeBind1(1, 5, "CORE.METADATA.ACTIONS.COPY_TO_CLIPBOARD"))("matTooltipDisabled", ctx_r6.isEditable);
    ɵngcc0.ɵɵattribute("data-automation-id", "card-textitem-value-" + ctx_r6.property.key);
} }
function CardViewTextItemComponent_div_2_textarea_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "textarea", 14);
} if (rf & 2) {
    const ctx_r7 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("matTextareaAutosize", true)("matAutosizeMaxRows", 1)("matAutosizeMaxRows", 5)("placeholder", ctx_r7.property.default)("formControl", ctx_r7.textInput);
    ɵngcc0.ɵɵattribute("data-automation-id", "card-textitem-value-" + ctx_r7.property.key);
} }
function CardViewTextItemComponent_div_2_button_4_Template(rf, ctx) { if (rf & 1) {
    const _r13 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 15);
    ɵngcc0.ɵɵlistener("click", function CardViewTextItemComponent_div_2_button_4_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r13); const ctx_r12 = ɵngcc0.ɵɵnextContext(2); return ctx_r12.clearValue(); });
    ɵngcc0.ɵɵelementStart(1, "mat-icon");
    ɵngcc0.ɵɵtext(2, "cancel");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} }
function CardViewTextItemComponent_div_2_mat_icon_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-icon", 16);
    ɵngcc0.ɵɵpipe(1, "translate");
    ɵngcc0.ɵɵtext(2, "mode_edit");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵattribute("title", ɵngcc0.ɵɵpipeBind1(1, 1, "CORE.METADATA.ACTIONS.EDIT"));
} }
const _c0 = function (a0) { return { "adf-property-read-only": a0 }; };
function CardViewTextItemComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelementStart(1, "mat-form-field", 8);
    ɵngcc0.ɵɵtemplate(2, CardViewTextItemComponent_div_2_input_2_Template, 2, 7, "input", 9);
    ɵngcc0.ɵɵtemplate(3, CardViewTextItemComponent_div_2_textarea_3_Template, 1, 6, "textarea", 10);
    ɵngcc0.ɵɵtemplate(4, CardViewTextItemComponent_div_2_button_4_Template, 3, 0, "button", 11);
    ɵngcc0.ɵɵtemplate(5, CardViewTextItemComponent_div_2_mat_icon_5_Template, 3, 3, "mat-icon", 12);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(6, _c0, !ctx_r1.isEditable))("floatLabel", "never");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r1.property.multiline);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1.property.multiline);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1.isEditable);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1.isEditable);
} }
function CardViewTextItemComponent_mat_error_3_li_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "li");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const error_r15 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind2(2, 1, error_r15.message, error_r15));
} }
function CardViewTextItemComponent_mat_error_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-error", 17);
    ɵngcc0.ɵɵelementStart(1, "ul");
    ɵngcc0.ɵɵtemplate(2, CardViewTextItemComponent_mat_error_3_li_2_Template, 3, 4, "li", 18);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵattribute("data-automation-id", "card-textitem-error-" + ctx_r2.property.key);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r2.errors);
} }
function CardViewTextItemComponent_div_4_mat_chip_3_mat_icon_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-icon", 26);
    ɵngcc0.ɵɵtext(1, "cancel");
    ɵngcc0.ɵɵelementEnd();
} }
function CardViewTextItemComponent_div_4_mat_chip_3_Template(rf, ctx) { if (rf & 1) {
    const _r23 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "mat-chip", 24);
    ɵngcc0.ɵɵlistener("removed", function CardViewTextItemComponent_div_4_mat_chip_3_Template_mat_chip_removed_0_listener() { ɵngcc0.ɵɵrestoreView(_r23); const idx_r20 = ctx.index; const ctx_r22 = ɵngcc0.ɵɵnextContext(2); return ctx_r22.removeValueFromList(idx_r20); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵtemplate(2, CardViewTextItemComponent_div_4_mat_chip_3_mat_icon_2_Template, 2, 0, "mat-icon", 25);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const propertyValue_r19 = ctx.$implicit;
    const ctx_r17 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("removable", ctx_r17.isEditable);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", propertyValue_r19, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r17.isEditable);
} }
function CardViewTextItemComponent_div_4_mat_form_field_4_Template(rf, ctx) { if (rf & 1) {
    const _r25 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "mat-form-field", 27);
    ɵngcc0.ɵɵelementStart(1, "input", 28);
    ɵngcc0.ɵɵlistener("matChipInputTokenEnd", function CardViewTextItemComponent_div_4_mat_form_field_4_Template_input_matChipInputTokenEnd_1_listener($event) { ɵngcc0.ɵɵrestoreView(_r25); const ctx_r24 = ɵngcc0.ɵɵnextContext(2); return ctx_r24.addValueToList($event); });
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "mat-icon", 16);
    ɵngcc0.ɵɵtext(4, "mode_edit");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵnextContext();
    const _r16 = ɵngcc0.ɵɵreference(2);
    const ctx_r18 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(8, _c0, !ctx_r18.isEditable))("floatLabel", "never");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("placeholder", ctx_r18.editedValue ? "" : ɵngcc0.ɵɵpipeBind1(2, 6, ctx_r18.property.default))("matChipInputFor", _r16)("matChipInputAddOnBlur", true);
    ɵngcc0.ɵɵattribute("data-automation-id", "card-textitem-editchipinput-" + ctx_r18.property.key);
} }
function CardViewTextItemComponent_div_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 19);
    ɵngcc0.ɵɵelementStart(1, "mat-chip-list", 20, 21);
    ɵngcc0.ɵɵtemplate(3, CardViewTextItemComponent_div_4_mat_chip_3_Template, 3, 3, "mat-chip", 22);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(4, CardViewTextItemComponent_div_4_mat_form_field_4_Template, 5, 10, "mat-form-field", 23);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r3.editedValue);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r3.isEditable);
} }
function CardViewTextItemComponent_div_5_button_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "button", 33);
    ɵngcc0.ɵɵpipe(1, "translate");
    ɵngcc0.ɵɵelementStart(2, "mat-icon", 34);
    ɵngcc0.ɵɵtext(3);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r26 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵattribute("title", ɵngcc0.ɵɵpipeBind1(1, 3, "CORE.METADATA.ACTIONS.EDIT"))("data-automation-id", "card-textitem-clickable-icon-" + ctx_r26.property.key);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate(ctx_r26.property == null ? null : ctx_r26.property.icon);
} }
const _c1 = function (a0) { return { "adf-textitem-clickable-value": a0 }; };
function CardViewTextItemComponent_div_5_Template(rf, ctx) { if (rf & 1) {
    const _r28 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 29);
    ɵngcc0.ɵɵlistener("click", function CardViewTextItemComponent_div_5_Template_div_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r28); const ctx_r27 = ɵngcc0.ɵɵnextContext(); return ctx_r27.clicked(); });
    ɵngcc0.ɵɵelementStart(1, "mat-form-field", 30);
    ɵngcc0.ɵɵelementStart(2, "input", 31);
    ɵngcc0.ɵɵlistener("ngModelChange", function CardViewTextItemComponent_div_5_Template_input_ngModelChange_2_listener($event) { ɵngcc0.ɵɵrestoreView(_r28); const ctx_r29 = ɵngcc0.ɵɵnextContext(); return ctx_r29.editedValue = $event; })("blur", function CardViewTextItemComponent_div_5_Template_input_blur_2_listener() { ɵngcc0.ɵɵrestoreView(_r28); const ctx_r30 = ɵngcc0.ɵɵnextContext(); return ctx_r30.update(); })("keydown.enter", function CardViewTextItemComponent_div_5_Template_input_keydown_enter_2_listener() { ɵngcc0.ɵɵrestoreView(_r28); const ctx_r31 = ɵngcc0.ɵɵnextContext(); return ctx_r31.update(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(3, CardViewTextItemComponent_div_5_button_3_Template, 4, 5, "button", 32);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(10, _c0, !ctx_r4.isEditable));
    ɵngcc0.ɵɵattribute("data-automation-id", "card-textitem-toggle-" + ctx_r4.property.key);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("floatLabel", "never");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("type", ctx_r4.property.inputType)("ngClass", ɵngcc0.ɵɵpureFunction1(12, _c1, !ctx_r4.isEditable))("placeholder", ctx_r4.property.default)("ngModel", ctx_r4.editedValue)("disabled", !ctx_r4.isEditable);
    ɵngcc0.ɵɵattribute("data-automation-id", "card-textitem-value-" + ctx_r4.property.key);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r4.showClickableIcon);
} }
function CardViewTextItemComponent_div_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelementStart(1, "span", 35);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵpipe(3, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r5 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(3, 1, ctx_r5.property.default));
} }
export const DEFAULT_SEPARATOR = ', ';
const templateTypes = {
    clickableTemplate: 'clickableTemplate',
    multilineTemplate: 'multilineTemplate',
    chipsTemplate: 'chipsTemplate',
    emptyTemplate: 'emptyTemplate',
    defaultTemplate: 'defaultTemplate'
};
export class CardViewTextItemComponent extends BaseCardView {
    constructor(cardViewUpdateService, clipboardService, translateService, cd) {
        super(cardViewUpdateService);
        this.clipboardService = clipboardService;
        this.translateService = translateService;
        this.cd = cd;
        this.editable = false;
        this.displayEmpty = true;
        this.copyToClipboardAction = true;
        this.useChipsForMultiValueProperty = true;
        this.multiValueSeparator = DEFAULT_SEPARATOR;
        this.textInput = new FormControl();
        this.onDestroy$ = new Subject();
    }
    ngOnChanges(changes) {
        if (changes.property && changes.property.firstChange) {
            this.textInput.valueChanges
                .pipe(filter(textInputValue => textInputValue !== this.editedValue && textInputValue !== null), debounceTime(50), takeUntil(this.onDestroy$))
                .subscribe(textInputValue => {
                this.editedValue = textInputValue;
                this.update();
            });
        }
        this.resetValue();
        this.setTemplateType();
        if (changes.editable) {
            this.isEditable ? this.textInput.enable() : this.textInput.disable();
        }
    }
    setTemplateType() {
        if (this.showProperty || this.isEditable) {
            if (this.isClickable) {
                this.templateType = templateTypes.clickableTemplate;
            }
            else if (this.isChipViewEnabled) {
                this.templateType = templateTypes.chipsTemplate;
            }
            else {
                this.templateType = templateTypes.defaultTemplate;
            }
        }
        else {
            this.templateType = templateTypes.emptyTemplate;
        }
    }
    resetValue() {
        if (this.isChipViewEnabled) {
            this.editedValue = this.property.value ? Array.from(this.property.value) : [];
        }
        else {
            this.editedValue = this.property.displayValue;
            this.textInput.setValue(this.editedValue);
        }
        this.resetErrorMessages();
    }
    resetErrorMessages() {
        this.errors = [];
    }
    update() {
        if (this.property.isValid(this.editedValue)) {
            this.property.value = this.prepareValueForUpload(this.property, this.editedValue);
            this.cardViewUpdateService.update(Object.assign({}, this.property), this.property.value);
            this.resetErrorMessages();
        }
        else {
            this.errors = this.property.getValidationErrors(this.editedValue);
        }
    }
    prepareValueForUpload(property, value) {
        if (property.multivalued && typeof value === 'string') {
            const listOfValues = value.split(this.multiValueSeparator.trim()).map((item) => item.trim());
            return listOfValues;
        }
        return value;
    }
    removeValueFromList(itemIndex) {
        if (Array.isArray(this.editedValue)) {
            this.editedValue.splice(itemIndex, 1);
            this.update();
            this.cd.detectChanges();
        }
    }
    addValueToList(newListItem) {
        const chipInput = newListItem.input;
        const chipValue = newListItem.value.trim() || '';
        if (typeof this.editedValue !== 'string') {
            if (chipValue) {
                this.editedValue.push(chipValue);
                this.update();
            }
            if (chipInput) {
                chipInput.value = '';
            }
        }
    }
    clicked() {
        if (typeof this.property.clickCallBack === 'function') {
            this.property.clickCallBack();
        }
        else {
            this.cardViewUpdateService.clicked(this.property);
        }
    }
    clearValue() {
        this.editedValue = '';
        this.textInput.setValue('');
        this.update();
    }
    copyToClipboard(valueToCopy) {
        if (this.copyToClipboard) {
            const clipboardMessage = this.translateService.instant('CORE.METADATA.ACCESSIBILITY.COPY_TO_CLIPBOARD_MESSAGE');
            this.clipboardService.copyContentToClipboard(valueToCopy, clipboardMessage);
        }
    }
    ngOnDestroy() {
        this.onDestroy$.next(true);
        this.onDestroy$.complete();
    }
    get showProperty() {
        return this.displayEmpty || !this.property.isEmpty();
    }
    get showClickableIcon() {
        return this.hasIcon && this.editable;
    }
    get isEditable() {
        return this.editable && this.property.editable;
    }
    get isClickable() {
        return this.property.clickable;
    }
    get hasIcon() {
        return !!this.property.icon;
    }
    get hasErrors() {
        var _a, _b;
        return (_b = (!!((_a = this.errors) === null || _a === void 0 ? void 0 : _a.length))) !== null && _b !== void 0 ? _b : false;
    }
    get isChipViewEnabled() {
        return this.property.multivalued && this.useChipsForMultiValueProperty;
    }
}
CardViewTextItemComponent.ɵfac = function CardViewTextItemComponent_Factory(t) { return new (t || CardViewTextItemComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.CardViewUpdateService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.ClipboardService), ɵngcc0.ɵɵdirectiveInject(ɵngcc3.TranslationService), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ChangeDetectorRef)); };
CardViewTextItemComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CardViewTextItemComponent, selectors: [["adf-card-view-textitem"]], inputs: { editable: "editable", displayEmpty: "displayEmpty", copyToClipboardAction: "copyToClipboardAction", useChipsForMultiValueProperty: "useChipsForMultiValueProperty", multiValueSeparator: "multiValueSeparator" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature, ɵngcc0.ɵɵNgOnChangesFeature], decls: 7, vars: 6, consts: [[3, "ngSwitch"], ["class", "adf-property-label", 4, "ngIf"], [4, "ngSwitchDefault"], ["class", "adf-textitem-editable-error", 4, "ngIf"], ["class", "adf-property-field adf-textitem-chip-list-container", 4, "ngSwitchCase"], ["role", "button", "class", "adf-textitem-clickable", "fxLayout", "row", "fxLayoutAlign", "space-between center", 3, "ngClass", "click", 4, "ngSwitchCase"], [4, "ngSwitchCase"], [1, "adf-property-label"], [1, "adf-property-field", "adf-card-textitem-field", 3, "ngClass", "floatLabel"], ["matInput", "", "class", "adf-property-value", "matTooltipShowDelay", "1000", 3, "placeholder", "formControl", "matTooltip", "matTooltipDisabled", "dblclick", 4, "ngIf"], ["matInput", "", "class", "adf-property-value", 3, "matTextareaAutosize", "matAutosizeMaxRows", "placeholder", "formControl", 4, "ngIf"], ["mat-button", "", "matSuffix", "", "class", "adf-property-clear-value", "mat-icon-button", "", "aria-label", "Clear", 3, "click", 4, "ngIf"], ["matSuffix", "", "class", "adf-textitem-edit-icon", 4, "ngIf"], ["matInput", "", "matTooltipShowDelay", "1000", 1, "adf-property-value", 3, "placeholder", "formControl", "matTooltip", "matTooltipDisabled", "dblclick"], ["matInput", "", 1, "adf-property-value", 3, "matTextareaAutosize", "matAutosizeMaxRows", "placeholder", "formControl"], ["mat-button", "", "matSuffix", "", "mat-icon-button", "", "aria-label", "Clear", 1, "adf-property-clear-value", 3, "click"], ["matSuffix", "", 1, "adf-textitem-edit-icon"], [1, "adf-textitem-editable-error"], [4, "ngFor", "ngForOf"], [1, "adf-property-field", "adf-textitem-chip-list-container"], [1, "adf-textitem-chip-list"], ["chipList", ""], [3, "removable", "removed", 4, "ngFor", "ngForOf"], ["class", "adf-property-field adf-textitem-chip-list-input", 3, "ngClass", "floatLabel", 4, "ngIf"], [3, "removable", "removed"], ["matChipRemove", "", 4, "ngIf"], ["matChipRemove", ""], [1, "adf-property-field", "adf-textitem-chip-list-input", 3, "ngClass", "floatLabel"], ["matInput", "", 1, "adf-property-value", 3, "placeholder", "matChipInputFor", "matChipInputAddOnBlur", "matChipInputTokenEnd"], ["role", "button", "fxLayout", "row", "fxLayoutAlign", "space-between center", 1, "adf-textitem-clickable", 3, "ngClass", "click"], [1, "adf-property-field", "adf-card-textitem-field", 3, "floatLabel"], ["matInput", "", 1, "adf-property-value", 3, "type", "ngClass", "placeholder", "ngModel", "disabled", "ngModelChange", "blur", "keydown.enter"], ["mat-icon-button", "", "matSuffix", "", "fxFlex", "0 0 auto", "class", "adf-textitem-action", 4, "ngIf"], ["mat-icon-button", "", "matSuffix", "", "fxFlex", "0 0 auto", 1, "adf-textitem-action"], [1, "adf-textitem-icon"], [1, "adf-textitem-default-value"]], template: function CardViewTextItemComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, CardViewTextItemComponent_div_1_Template, 3, 4, "div", 1);
        ɵngcc0.ɵɵtemplate(2, CardViewTextItemComponent_div_2_Template, 6, 8, "div", 2);
        ɵngcc0.ɵɵtemplate(3, CardViewTextItemComponent_mat_error_3_Template, 3, 2, "mat-error", 3);
        ɵngcc0.ɵɵtemplate(4, CardViewTextItemComponent_div_4_Template, 5, 2, "div", 4);
        ɵngcc0.ɵɵtemplate(5, CardViewTextItemComponent_div_5_Template, 4, 14, "div", 5);
        ɵngcc0.ɵɵtemplate(6, CardViewTextItemComponent_div_6_Template, 4, 3, "div", 6);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngSwitch", ctx.templateType);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showProperty || ctx.isEditable);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isEditable && ctx.hasErrors);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "chipsTemplate");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "clickableTemplate");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "emptyTemplate");
    } }, directives: [ɵngcc4.NgSwitch, ɵngcc4.NgIf, ɵngcc4.NgSwitchDefault, ɵngcc4.NgSwitchCase, ɵngcc5.MatFormField, ɵngcc4.NgClass, ɵngcc6.DefaultClassDirective, ɵngcc7.MatInput, ɵngcc8.DefaultValueAccessor, ɵngcc8.NgControlStatus, ɵngcc8.FormControlDirective, ɵngcc9.MatTooltip, ɵngcc7.MatTextareaAutosize, ɵngcc10.MatButton, ɵngcc5.MatSuffix, ɵngcc11.MatIcon, ɵngcc5.MatError, ɵngcc4.NgForOf, ɵngcc12.MatChipList, ɵngcc12.MatChip, ɵngcc12.MatChipRemove, ɵngcc12.MatChipInput, ɵngcc13.DefaultLayoutDirective, ɵngcc13.DefaultLayoutAlignDirective, ɵngcc8.NgModel, ɵngcc13.DefaultFlexDirective], pipes: [ɵngcc14.TranslatePipe], encapsulation: 2 });
CardViewTextItemComponent.ctorParameters = () => [
    { type: CardViewUpdateService },
    { type: ClipboardService },
    { type: TranslationService },
    { type: ChangeDetectorRef }
];
CardViewTextItemComponent.propDecorators = {
    editable: [{ type: Input }],
    displayEmpty: [{ type: Input }],
    copyToClipboardAction: [{ type: Input }],
    useChipsForMultiValueProperty: [{ type: Input }],
    multiValueSeparator: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CardViewTextItemComponent, [{
        type: Component,
        args: [{
                selector: 'adf-card-view-textitem',
                template: "<div [ngSwitch]=\"templateType\">\n    <div class=\"adf-property-label\"\n         [attr.data-automation-id]=\"'card-textitem-label-' + property.key\"\n         *ngIf=\"showProperty || isEditable\">\n        {{ property.label | translate }}\n    </div>\n\n    <div *ngSwitchDefault>\n        <mat-form-field class=\"adf-property-field adf-card-textitem-field\"\n                        [ngClass]=\"{ 'adf-property-read-only': !isEditable }\"\n                        [floatLabel]=\"'never'\">\n            <input matInput\n                   *ngIf=\"!property.multiline\"\n                   class=\"adf-property-value\"\n                   [placeholder]=\"property.default\"\n                   [formControl]=\"textInput\"\n                   (dblclick)=\"copyToClipboard(property.displayValue)\"\n                   matTooltipShowDelay=\"1000\"\n                   [matTooltip]=\"'CORE.METADATA.ACTIONS.COPY_TO_CLIPBOARD' | translate\"\n                   [matTooltipDisabled]=\"isEditable\"\n                   [attr.data-automation-id]=\"'card-textitem-value-' + property.key\">\n            <textarea matInput\n                      *ngIf=\"property.multiline\"\n                      [matTextareaAutosize]=\"true\"\n                      [matAutosizeMaxRows]=\"1\"\n                      [matAutosizeMaxRows]=\"5\"\n                      class=\"adf-property-value\"\n                      [placeholder]=\"property.default\"\n                      [formControl]=\"textInput\"\n                      [attr.data-automation-id]=\"'card-textitem-value-' + property.key\"></textarea>\n            <button mat-button\n                    matSuffix\n                    class=\"adf-property-clear-value\"\n                    *ngIf=\"isEditable\"\n                    mat-icon-button\n                    aria-label=\"Clear\"\n                    (click)=\"clearValue()\">\n                <mat-icon>cancel</mat-icon>\n            </button>\n            <mat-icon matSuffix\n                      *ngIf=\"isEditable\"\n                      [attr.title]=\"'CORE.METADATA.ACTIONS.EDIT' | translate\"\n                      class=\"adf-textitem-edit-icon\">mode_edit</mat-icon>\n\n        </mat-form-field>\n    </div>\n\n    <mat-error [attr.data-automation-id]=\"'card-textitem-error-' + property.key\"\n               class=\"adf-textitem-editable-error\"\n               *ngIf=\"isEditable && hasErrors\">\n        <ul>\n            <li *ngFor=\"let error of errors\">{{ error.message | translate: error }}</li>\n        </ul>\n    </mat-error>\n\n    <div *ngSwitchCase=\"'chipsTemplate'\"\n         class=\"adf-property-field adf-textitem-chip-list-container\">\n        <mat-chip-list #chipList\n                       class=\"adf-textitem-chip-list\">\n            <mat-chip *ngFor=\"let propertyValue of editedValue; let idx = index\"\n                      [removable]=\"isEditable\"\n                      (removed)=\"removeValueFromList(idx)\">\n                {{ propertyValue }}\n                <mat-icon *ngIf=\"isEditable\"\n                          matChipRemove>cancel</mat-icon>\n            </mat-chip>\n        </mat-chip-list>\n\n        <mat-form-field *ngIf=\"isEditable\"\n                        class=\"adf-property-field adf-textitem-chip-list-input\"\n                        [ngClass]=\"{ 'adf-property-read-only': !isEditable }\"\n                        [floatLabel]=\"'never'\">\n            <input matInput\n                   class=\"adf-property-value\"\n                   [placeholder]=\"editedValue ? '' : property.default | translate\"\n                   [matChipInputFor]=\"chipList\"\n                   [matChipInputAddOnBlur]=\"true\"\n                   (matChipInputTokenEnd)=\"addValueToList($event)\"\n                   [attr.data-automation-id]=\"'card-textitem-editchipinput-' + property.key\">\n            <mat-icon matSuffix\n                      class=\"adf-textitem-edit-icon\">mode_edit</mat-icon>\n        </mat-form-field>\n    </div>\n\n    <div *ngSwitchCase=\"'clickableTemplate'\"\n         role=\"button\"\n         class=\"adf-textitem-clickable\"\n         [ngClass]=\"{ 'adf-property-read-only': !isEditable }\"\n         [attr.data-automation-id]=\"'card-textitem-toggle-' + property.key\"\n         (click)=\"clicked()\"\n         fxLayout=\"row\"\n         fxLayoutAlign=\"space-between center\">\n        <mat-form-field class=\"adf-property-field adf-card-textitem-field\"\n                        [floatLabel]=\"'never'\">\n            <input matInput\n                   [type]=property.inputType\n                   class=\"adf-property-value\"\n                   [ngClass]=\"{ 'adf-textitem-clickable-value': !isEditable }\"\n                   [placeholder]=\"property.default\"\n                   [(ngModel)]=\"editedValue\"\n                   (blur)=\"update()\"\n                   (keydown.enter)=\"update()\"\n                   [disabled]=\"!isEditable\"\n                   [attr.data-automation-id]=\"'card-textitem-value-' + property.key\">\n            <button mat-icon-button\n                    matSuffix\n                    fxFlex=\"0 0 auto\"\n                    *ngIf=\"showClickableIcon\"\n                    class=\"adf-textitem-action\"\n                    [attr.title]=\"'CORE.METADATA.ACTIONS.EDIT' | translate\"\n                    [attr.data-automation-id]=\"'card-textitem-clickable-icon-' + property.key\">\n                <mat-icon class=\"adf-textitem-icon\">{{ property?.icon }}</mat-icon>\n            </button>\n        </mat-form-field>\n\n    </div>\n\n    <div *ngSwitchCase=\"'emptyTemplate'\">\n        <span class=\"adf-textitem-default-value\">{{ property.default | translate }}</span>\n    </div>\n</div>\n"
            }]
    }], function () { return [{ type: ɵngcc1.CardViewUpdateService }, { type: ɵngcc2.ClipboardService }, { type: ɵngcc3.TranslationService }, { type: ɵngcc0.ChangeDetectorRef }]; }, { editable: [{
            type: Input
        }], displayEmpty: [{
            type: Input
        }], copyToClipboardAction: [{
            type: Input
        }], useChipsForMultiValueProperty: [{
            type: Input
        }], multiValueSeparator: [{
            type: Input
        }] }); })();
