// @alfresco/adf-core/esm2015/userinfo/components/user-info.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Input, ViewEncapsulation, ViewChild } from '@angular/core';
import { AuthenticationService } from '../../services/authentication.service';
import { BpmUserService } from '../../services/bpm-user.service';
import { EcmUserService } from '../../services/ecm-user.service';
import { IdentityUserService } from '../../services/identity-user.service';
import { of } from 'rxjs';
import { MatMenuTrigger } from '@angular/material/menu';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../services/ecm-user.service';
import * as ɵngcc2 from '../../services/bpm-user.service';
import * as ɵngcc3 from '../../services/identity-user.service';
import * as ɵngcc4 from '../../services/authentication.service';
import * as ɵngcc5 from '@angular/common';
import * as ɵngcc6 from '@angular/material/button';
import * as ɵngcc7 from '@angular/material/menu';
import * as ɵngcc8 from '@angular/material/tabs';
import * as ɵngcc9 from '@angular/material/card';
import * as ɵngcc10 from '../../pipes/full-name.pipe';
import * as ɵngcc11 from '../../pipes/user-initial.pipe';
import * as ɵngcc12 from '@ngx-translate/core';

function UserInfoComponent_div_0_ng_container_1_span_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 15);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "fullName");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const identityUser_r12 = ctx.ngIf;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, identityUser_r12));
} }
function UserInfoComponent_div_0_ng_container_1_ng_template_3_span_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 18);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "fullName");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ecmUser_r16 = ctx.ngIf;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, ecmUser_r16));
} }
function UserInfoComponent_div_0_ng_container_1_ng_template_3_ng_template_2_span_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 20);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "fullName");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const bpmUser_r18 = ctx.ngIf;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(2, 1, bpmUser_r18));
} }
function UserInfoComponent_div_0_ng_container_1_ng_template_3_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, UserInfoComponent_div_0_ng_container_1_ng_template_3_ng_template_2_span_0_Template, 3, 3, "span", 19);
    ɵngcc0.ɵɵpipe(1, "async");
} if (rf & 2) {
    const ctx_r15 = ɵngcc0.ɵɵnextContext(4);
    ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(1, 1, ctx_r15.bpmUser$));
} }
function UserInfoComponent_div_0_ng_container_1_ng_template_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, UserInfoComponent_div_0_ng_container_1_ng_template_3_span_0_Template, 3, 3, "span", 16);
    ɵngcc0.ɵɵpipe(1, "async");
    ɵngcc0.ɵɵtemplate(2, UserInfoComponent_div_0_ng_container_1_ng_template_3_ng_template_2_Template, 2, 3, "ng-template", null, 17, ɵngcc0.ɵɵtemplateRefExtractor);
} if (rf & 2) {
    const _r14 = ɵngcc0.ɵɵreference(3);
    const ctx_r11 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(1, 2, ctx_r11.ecmUser$))("ngIfElse", _r14);
} }
function UserInfoComponent_div_0_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, UserInfoComponent_div_0_ng_container_1_span_1_Template, 3, 3, "span", 13);
    ɵngcc0.ɵɵpipe(2, "async");
    ɵngcc0.ɵɵtemplate(3, UserInfoComponent_div_0_ng_container_1_ng_template_3_Template, 4, 4, "ng-template", null, 14, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const _r10 = ɵngcc0.ɵɵreference(4);
    const ctx_r1 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(2, 2, ctx_r1.identityUser$))("ngIfElse", _r10);
} }
function UserInfoComponent_div_0_div_4_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵelementStart(1, "div", 24);
    ɵngcc0.ɵɵelement(2, "img", 25);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const avatarId_r23 = ctx.ngIf;
    const ctx_r20 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("src", ctx_r20.getEcmAvatar(avatarId_r23), ɵngcc0.ɵɵsanitizeUrl);
} }
function UserInfoComponent_div_0_div_4_ng_template_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 26);
    ɵngcc0.ɵɵpipe(1, "usernameInitials");
} if (rf & 2) {
    const identityUser_r19 = ɵngcc0.ɵɵnextContext().ngIf;
    ɵngcc0.ɵɵproperty("innerHTML", ɵngcc0.ɵɵpipeBind2(1, 1, identityUser_r19, "adf-userinfo-pic"), ɵngcc0.ɵɵsanitizeHtml);
} }
function UserInfoComponent_div_0_div_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 21);
    ɵngcc0.ɵɵtemplate(1, UserInfoComponent_div_0_div_4_div_1_Template, 3, 1, "div", 22);
    ɵngcc0.ɵɵpipe(2, "async");
    ɵngcc0.ɵɵtemplate(3, UserInfoComponent_div_0_div_4_ng_template_3_Template, 2, 4, "ng-template", null, 23, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const _r21 = ɵngcc0.ɵɵreference(4);
    const ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    var tmp_0_0 = null;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", (tmp_0_0 = ɵngcc0.ɵɵpipeBind1(2, 2, ctx_r2.ecmUser$)) == null ? null : tmp_0_0.avatarId)("ngIfElse", _r21);
} }
function UserInfoComponent_div_0_ng_template_6_div_0_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 24);
    ɵngcc0.ɵɵelement(1, "img", 25);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ecmUser_r28 = ɵngcc0.ɵɵnextContext().ngIf;
    const ctx_r29 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("src", ctx_r29.getEcmAvatar(ecmUser_r28.avatarId), ɵngcc0.ɵɵsanitizeUrl);
} }
function UserInfoComponent_div_0_ng_template_6_div_0_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 31);
    ɵngcc0.ɵɵpipe(1, "usernameInitials");
} if (rf & 2) {
    const ecmUser_r28 = ɵngcc0.ɵɵnextContext().ngIf;
    ɵngcc0.ɵɵproperty("outerHTML", ɵngcc0.ɵɵpipeBind2(1, 1, ecmUser_r28, "adf-userinfo-pic"), ɵngcc0.ɵɵsanitizeHtml);
} }
function UserInfoComponent_div_0_ng_template_6_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 29);
    ɵngcc0.ɵɵtemplate(1, UserInfoComponent_div_0_ng_template_6_div_0_div_1_Template, 2, 1, "div", 30);
    ɵngcc0.ɵɵtemplate(2, UserInfoComponent_div_0_ng_template_6_div_0_ng_template_2_Template, 2, 4, "ng-template", null, 23, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ecmUser_r28 = ctx.ngIf;
    const _r30 = ɵngcc0.ɵɵreference(3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ecmUser_r28.avatarId)("ngIfElse", _r30);
} }
function UserInfoComponent_div_0_ng_template_6_ng_template_2_div_0_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 24);
    ɵngcc0.ɵɵelement(1, "img", 25);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r36 = ɵngcc0.ɵɵnextContext(5);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("src", ctx_r36.getBpmUserImage(), ɵngcc0.ɵɵsanitizeUrl);
} }
function UserInfoComponent_div_0_ng_template_6_ng_template_2_div_0_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 31);
    ɵngcc0.ɵɵpipe(1, "usernameInitials");
} if (rf & 2) {
    const bpmUser_r35 = ɵngcc0.ɵɵnextContext().ngIf;
    ɵngcc0.ɵɵproperty("outerHTML", ɵngcc0.ɵɵpipeBind2(1, 1, bpmUser_r35, "adf-userinfo-pic"), ɵngcc0.ɵɵsanitizeHtml);
} }
function UserInfoComponent_div_0_ng_template_6_ng_template_2_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 33);
    ɵngcc0.ɵɵtemplate(1, UserInfoComponent_div_0_ng_template_6_ng_template_2_div_0_div_1_Template, 2, 1, "div", 30);
    ɵngcc0.ɵɵtemplate(2, UserInfoComponent_div_0_ng_template_6_ng_template_2_div_0_ng_template_2_Template, 2, 4, "ng-template", null, 23, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const bpmUser_r35 = ctx.ngIf;
    const _r37 = ɵngcc0.ɵɵreference(3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", bpmUser_r35.pictureId)("ngIfElse", _r37);
} }
function UserInfoComponent_div_0_ng_template_6_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, UserInfoComponent_div_0_ng_template_6_ng_template_2_div_0_Template, 4, 2, "div", 32);
    ɵngcc0.ɵɵpipe(1, "async");
} if (rf & 2) {
    const ctx_r27 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(1, 1, ctx_r27.bpmUser$));
} }
function UserInfoComponent_div_0_ng_template_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, UserInfoComponent_div_0_ng_template_6_div_0_Template, 4, 2, "div", 27);
    ɵngcc0.ɵɵpipe(1, "async");
    ɵngcc0.ɵɵtemplate(2, UserInfoComponent_div_0_ng_template_6_ng_template_2_Template, 2, 3, "ng-template", null, 28, ɵngcc0.ɵɵtemplateRefExtractor);
} if (rf & 2) {
    const _r26 = ɵngcc0.ɵɵreference(3);
    const ctx_r4 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(1, 2, ctx_r4.ecmUser$))("ngIfElse", _r26);
} }
function UserInfoComponent_div_0_mat_tab_13_mat_card_2_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 46);
    ɵngcc0.ɵɵelement(1, "img", 47);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ecmUser_r41 = ɵngcc0.ɵɵnextContext().ngIf;
    const ctx_r42 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("src", ctx_r42.getEcmAvatar(ecmUser_r41.avatarId), ɵngcc0.ɵɵsanitizeUrl);
} }
function UserInfoComponent_div_0_mat_tab_13_mat_card_2_ng_template_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 31);
    ɵngcc0.ɵɵpipe(1, "usernameInitials");
} if (rf & 2) {
    const ecmUser_r41 = ɵngcc0.ɵɵnextContext().ngIf;
    ɵngcc0.ɵɵproperty("outerHTML", ɵngcc0.ɵɵpipeBind2(1, 1, ecmUser_r41, "adf-userinfo-profile-initials adf-hide-small"), ɵngcc0.ɵɵsanitizeHtml);
} }
function UserInfoComponent_div_0_mat_tab_13_mat_card_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-card", 36);
    ɵngcc0.ɵɵelementStart(1, "mat-card-header", 37);
    ɵngcc0.ɵɵtemplate(2, UserInfoComponent_div_0_mat_tab_13_mat_card_2_div_2_Template, 2, 1, "div", 38);
    ɵngcc0.ɵɵtemplate(3, UserInfoComponent_div_0_mat_tab_13_mat_card_2_ng_template_3_Template, 2, 4, "ng-template", null, 23, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementStart(5, "div", 39);
    ɵngcc0.ɵɵtext(6);
    ɵngcc0.ɵɵpipe(7, "fullName");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(8, "mat-card-content");
    ɵngcc0.ɵɵelementStart(9, "div", 40);
    ɵngcc0.ɵɵelementStart(10, "div", 41);
    ɵngcc0.ɵɵelementStart(11, "span", 42);
    ɵngcc0.ɵɵtext(12);
    ɵngcc0.ɵɵpipe(13, "fullName");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(14, "span", 43);
    ɵngcc0.ɵɵtext(15);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(16, "div", 41);
    ɵngcc0.ɵɵelementStart(17, "span", 44);
    ɵngcc0.ɵɵtext(18);
    ɵngcc0.ɵɵpipe(19, "translate");
    ɵngcc0.ɵɵelementStart(20, "span", 45);
    ɵngcc0.ɵɵtext(21);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ecmUser_r41 = ctx.ngIf;
    const _r43 = ɵngcc0.ɵɵreference(4);
    const ctx_r40 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵstyleProp("background-image", "url(" + ctx_r40.ecmBackgroundImage + ")");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ecmUser_r41.avatarId)("ngIfElse", _r43);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(7, 9, ecmUser_r41));
    ɵngcc0.ɵɵadvance(6);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(13, 11, ecmUser_r41));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", ecmUser_r41.email, " ");
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(19, 13, "USER_PROFILE.LABELS.ECM.JOB_TITLE"), " ");
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", ecmUser_r41.jobTitle ? ecmUser_r41.jobTitle : "N/A", " ");
} }
function UserInfoComponent_div_0_mat_tab_13_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-tab", 34);
    ɵngcc0.ɵɵpipe(1, "translate");
    ɵngcc0.ɵɵtemplate(2, UserInfoComponent_div_0_mat_tab_13_mat_card_2_Template, 22, 15, "mat-card", 35);
    ɵngcc0.ɵɵpipe(3, "async");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r6 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵpropertyInterpolate("label", ɵngcc0.ɵɵpipeBind1(1, 2, "USER_PROFILE.TAB.CS"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(3, 4, ctx_r6.ecmUser$));
} }
function UserInfoComponent_div_0_mat_tab_14_mat_card_2_img_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "img", 55);
} if (rf & 2) {
    const ctx_r49 = ɵngcc0.ɵɵnextContext(4);
    ɵngcc0.ɵɵproperty("src", ctx_r49.getBpmUserImage(), ɵngcc0.ɵɵsanitizeUrl);
} }
function UserInfoComponent_div_0_mat_tab_14_mat_card_2_ng_template_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 31);
    ɵngcc0.ɵɵpipe(1, "usernameInitials");
} if (rf & 2) {
    const bpmUser_r48 = ɵngcc0.ɵɵnextContext().ngIf;
    ɵngcc0.ɵɵproperty("outerHTML", ɵngcc0.ɵɵpipeBind2(1, 1, bpmUser_r48, "adf-userinfo-profile-initials adf-hide-small"), ɵngcc0.ɵɵsanitizeHtml);
} }
function UserInfoComponent_div_0_mat_tab_14_mat_card_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-card", 36);
    ɵngcc0.ɵɵelementStart(1, "mat-card-header", 37);
    ɵngcc0.ɵɵtemplate(2, UserInfoComponent_div_0_mat_tab_14_mat_card_2_img_2_Template, 1, 1, "img", 49);
    ɵngcc0.ɵɵtemplate(3, UserInfoComponent_div_0_mat_tab_14_mat_card_2_ng_template_3_Template, 2, 4, "ng-template", null, 23, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementStart(5, "div", 50);
    ɵngcc0.ɵɵtext(6);
    ɵngcc0.ɵɵpipe(7, "fullName");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(8, "mat-card-content");
    ɵngcc0.ɵɵelementStart(9, "div", 40);
    ɵngcc0.ɵɵelementStart(10, "div", 41);
    ɵngcc0.ɵɵelementStart(11, "span", 51);
    ɵngcc0.ɵɵtext(12);
    ɵngcc0.ɵɵpipe(13, "fullName");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(14, "span", 52);
    ɵngcc0.ɵɵtext(15);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(16, "div", 41);
    ɵngcc0.ɵɵelementStart(17, "span", 53);
    ɵngcc0.ɵɵtext(18);
    ɵngcc0.ɵɵpipe(19, "translate");
    ɵngcc0.ɵɵelementStart(20, "span", 54);
    ɵngcc0.ɵɵtext(21);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const bpmUser_r48 = ctx.ngIf;
    const _r50 = ɵngcc0.ɵɵreference(4);
    const ctx_r47 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵstyleProp("background-image", "url(" + ctx_r47.bpmBackgroundImage + ")");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", bpmUser_r48.pictureId)("ngIfElse", _r50);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(7, 9, bpmUser_r48));
    ɵngcc0.ɵɵadvance(6);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(13, 11, bpmUser_r48));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", bpmUser_r48.email, " ");
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(19, 13, "USER_PROFILE.LABELS.BPM.TENANT"), " ");
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate(bpmUser_r48.tenantName ? bpmUser_r48.tenantName : "");
} }
function UserInfoComponent_div_0_mat_tab_14_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-tab", 48);
    ɵngcc0.ɵɵpipe(1, "translate");
    ɵngcc0.ɵɵtemplate(2, UserInfoComponent_div_0_mat_tab_14_mat_card_2_Template, 22, 15, "mat-card", 35);
    ɵngcc0.ɵɵpipe(3, "async");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r7 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵpropertyInterpolate("label", ɵngcc0.ɵɵpipeBind1(1, 2, "USER_PROFILE.TAB.PS"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(3, 4, ctx_r7.bpmUser$));
} }
function UserInfoComponent_div_0_mat_tab_15_mat_card_1_div_2_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 46);
    ɵngcc0.ɵɵelement(1, "img", 47);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ecmUser_r58 = ɵngcc0.ɵɵnextContext().ngIf;
    const ctx_r59 = ɵngcc0.ɵɵnextContext(4);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("src", ctx_r59.getEcmAvatar(ecmUser_r58.avatarId), ɵngcc0.ɵɵsanitizeUrl);
} }
function UserInfoComponent_div_0_mat_tab_15_mat_card_1_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵtemplate(1, UserInfoComponent_div_0_mat_tab_15_mat_card_1_div_2_div_1_Template, 2, 1, "div", 38);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ecmUser_r58 = ctx.ngIf;
    ɵngcc0.ɵɵnextContext();
    const _r56 = ɵngcc0.ɵɵreference(5);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ecmUser_r58.avatarId)("ngIfElse", _r56);
} }
function UserInfoComponent_div_0_mat_tab_15_mat_card_1_ng_template_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 31);
    ɵngcc0.ɵɵpipe(1, "usernameInitials");
} if (rf & 2) {
    const identityUser_r54 = ɵngcc0.ɵɵnextContext().ngIf;
    ɵngcc0.ɵɵproperty("outerHTML", ɵngcc0.ɵɵpipeBind2(1, 1, identityUser_r54, "adf-userinfo-profile-initials adf-hide-small"), ɵngcc0.ɵɵsanitizeHtml);
} }
function UserInfoComponent_div_0_mat_tab_15_mat_card_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-card", 36);
    ɵngcc0.ɵɵelementStart(1, "mat-card-header", 37);
    ɵngcc0.ɵɵtemplate(2, UserInfoComponent_div_0_mat_tab_15_mat_card_1_div_2_Template, 2, 2, "div", 2);
    ɵngcc0.ɵɵpipe(3, "async");
    ɵngcc0.ɵɵtemplate(4, UserInfoComponent_div_0_mat_tab_15_mat_card_1_ng_template_4_Template, 2, 4, "ng-template", null, 23, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementStart(6, "div", 57);
    ɵngcc0.ɵɵtext(7);
    ɵngcc0.ɵɵpipe(8, "fullName");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(9, "mat-card-content");
    ɵngcc0.ɵɵelementStart(10, "div", 40);
    ɵngcc0.ɵɵelementStart(11, "div", 41);
    ɵngcc0.ɵɵelementStart(12, "span", 58);
    ɵngcc0.ɵɵtext(13);
    ɵngcc0.ɵɵpipe(14, "fullName");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(15, "span", 59);
    ɵngcc0.ɵɵtext(16);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const identityUser_r54 = ctx.ngIf;
    const ctx_r53 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵstyleProp("background-image", "url(" + ctx_r53.bpmBackgroundImage + ")");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(3, 6, ctx_r53.ecmUser$));
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(8, 8, identityUser_r54));
    ɵngcc0.ɵɵadvance(6);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(14, 10, identityUser_r54));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", identityUser_r54.email, " ");
} }
function UserInfoComponent_div_0_mat_tab_15_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "mat-tab", 56);
    ɵngcc0.ɵɵtemplate(1, UserInfoComponent_div_0_mat_tab_15_mat_card_1_Template, 17, 12, "mat-card", 35);
    ɵngcc0.ɵɵpipe(2, "async");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r8 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(2, 1, ctx_r8.identityUser$));
} }
function UserInfoComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    const _r63 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 1);
    ɵngcc0.ɵɵlistener("keyup", function UserInfoComponent_div_0_Template_div_keyup_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r63); const ctx_r62 = ɵngcc0.ɵɵnextContext(); return ctx_r62.onKeyPress($event); });
    ɵngcc0.ɵɵtemplate(1, UserInfoComponent_div_0_ng_container_1_Template, 5, 4, "ng-container", 2);
    ɵngcc0.ɵɵelementStart(2, "button", 3);
    ɵngcc0.ɵɵelementStart(3, "div", 4);
    ɵngcc0.ɵɵtemplate(4, UserInfoComponent_div_0_div_4_Template, 5, 4, "div", 5);
    ɵngcc0.ɵɵpipe(5, "async");
    ɵngcc0.ɵɵtemplate(6, UserInfoComponent_div_0_ng_template_6_Template, 4, 4, "ng-template", null, 6, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(8, "mat-menu", 7, 8);
    ɵngcc0.ɵɵelementStart(10, "mat-tab-group", 9);
    ɵngcc0.ɵɵlistener("click", function UserInfoComponent_div_0_Template_mat_tab_group_click_10_listener($event) { ɵngcc0.ɵɵrestoreView(_r63); const ctx_r64 = ɵngcc0.ɵɵnextContext(); return ctx_r64.stopClosing($event); });
    ɵngcc0.ɵɵpipe(11, "async");
    ɵngcc0.ɵɵpipe(12, "async");
    ɵngcc0.ɵɵtemplate(13, UserInfoComponent_div_0_mat_tab_13_Template, 4, 6, "mat-tab", 10);
    ɵngcc0.ɵɵtemplate(14, UserInfoComponent_div_0_mat_tab_14_Template, 4, 6, "mat-tab", 11);
    ɵngcc0.ɵɵtemplate(15, UserInfoComponent_div_0_mat_tab_15_Template, 3, 3, "mat-tab", 12);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const _r3 = ɵngcc0.ɵɵreference(7);
    const _r5 = ɵngcc0.ɵɵreference(9);
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("adf-userinfo-name-right", ctx_r0.showOnRight);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.showName);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("matMenuTriggerFor", _r5);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(5, 14, ctx_r0.identityUser$))("ngIfElse", _r3);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵproperty("xPosition", ctx_r0.menuPositionX)("yPosition", ctx_r0.menuPositionY)("overlapTrigger", false);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵclassProp("adf-hide-tab", !ɵngcc0.ɵɵpipeBind1(11, 16, ctx_r0.bpmUser$) || !ɵngcc0.ɵɵpipeBind1(12, 18, ctx_r0.ecmUser$));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.mode === "CONTENT" || ctx_r0.mode === "ALL");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.mode === "PROCESS" || ctx_r0.mode === "ALL");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.mode === "SSO");
} }
export class UserInfoComponent {
    constructor(ecmUserService, bpmUserService, identityUserService, authService) {
        this.ecmUserService = ecmUserService;
        this.bpmUserService = bpmUserService;
        this.identityUserService = identityUserService;
        this.authService = authService;
        this.ecmBackgroundImage = './assets/images/ecm-background.png';
        this.bpmBackgroundImage = './assets/images/bpm-background.png';
        this.menuPositionX = 'after';
        this.menuPositionY = 'below';
        this.showName = true;
        this.namePosition = 'right';
    }
    ngOnInit() {
        this.getUserInfo();
    }
    getUserInfo() {
        if (this.authService.isOauth()) {
            this.loadIdentityUserInfo();
            this.mode = 'SSO';
            if (this.authService.isEcmLoggedIn()) {
                this.loadEcmUserInfo();
            }
        }
        else if (this.authService.isEcmLoggedIn() && this.authService.isBpmLoggedIn()) {
            this.loadEcmUserInfo();
            this.loadBpmUserInfo();
            this.mode = 'ALL';
        }
        else if (this.authService.isEcmLoggedIn()) {
            this.loadEcmUserInfo();
            this.mode = 'CONTENT';
        }
        else if (this.authService.isBpmLoggedIn()) {
            this.loadBpmUserInfo();
            this.mode = 'PROCESS';
        }
    }
    onKeyPress(event) {
        this.closeUserModal(event);
    }
    closeUserModal($event) {
        if ($event.keyCode === 27) {
            this.trigger.closeMenu();
        }
    }
    get isLoggedIn() {
        return this.authService.isLoggedIn();
    }
    loadEcmUserInfo() {
        this.ecmUser$ = this.ecmUserService.getCurrentUserInfo();
    }
    loadBpmUserInfo() {
        this.bpmUser$ = this.bpmUserService.getCurrentUserInfo();
    }
    loadIdentityUserInfo() {
        this.identityUser$ = of(this.identityUserService.getCurrentUserInfo());
    }
    stopClosing(event) {
        event.stopPropagation();
    }
    getEcmAvatar(avatarId) {
        return this.ecmUserService.getUserProfileImage(avatarId);
    }
    getBpmUserImage() {
        return this.bpmUserService.getCurrentUserProfileImage();
    }
    get showOnRight() {
        return this.namePosition === 'right';
    }
}
UserInfoComponent.ɵfac = function UserInfoComponent_Factory(t) { return new (t || UserInfoComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.EcmUserService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.BpmUserService), ɵngcc0.ɵɵdirectiveInject(ɵngcc3.IdentityUserService), ɵngcc0.ɵɵdirectiveInject(ɵngcc4.AuthenticationService)); };
UserInfoComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: UserInfoComponent, selectors: [["adf-userinfo"]], viewQuery: function UserInfoComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(MatMenuTrigger, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.trigger = _t.first);
    } }, inputs: { ecmBackgroundImage: "ecmBackgroundImage", bpmBackgroundImage: "bpmBackgroundImage", menuPositionX: "menuPositionX", menuPositionY: "menuPositionY", showName: "showName", namePosition: "namePosition" }, decls: 1, vars: 1, consts: [["id", "userinfo_container", "class", "adf-userinfo-container", 3, "adf-userinfo-name-right", "keyup", 4, "ngIf"], ["id", "userinfo_container", 1, "adf-userinfo-container", 3, "keyup"], [4, "ngIf"], ["mat-button", "", "data-automation-id", "adf-user-profile", 1, "adf-userinfo-menu_button", 3, "matMenuTriggerFor"], ["id", "user-profile", 1, "adf-userinfo-button-profile"], ["id", "identity-user-image", 4, "ngIf", "ngIfElse"], ["showBpmAndEcmUserImage", ""], ["id", "user-profile-lists", 1, "adf-userinfo-menu", 3, "xPosition", "yPosition", "overlapTrigger"], ["menu", "matMenu"], ["id", "tab-group-env", "selectedIndex", "0", 1, "adf-userinfo-tab", 3, "click"], ["id", "ecm-panel", 3, "label", 4, "ngIf"], ["id", "bpm-panel", 3, "label", 4, "ngIf"], ["id", "identity-panel", 4, "ngIf"], ["id", "adf-userinfo-identity-name-display", "class", "adf-userinfo-name", 4, "ngIf", "ngIfElse"], ["showBpmAndEcmUserFullNames", ""], ["id", "adf-userinfo-identity-name-display", 1, "adf-userinfo-name"], ["id", "adf-userinfo-ecm-name-display", "class", "adf-userinfo-name", 4, "ngIf", "ngIfElse"], ["showBpmUserFullName", ""], ["id", "adf-userinfo-ecm-name-display", 1, "adf-userinfo-name"], ["id", "adf-userinfo-bpm-name-display", "class", "adf-userinfo-name", 4, "ngIf"], ["id", "adf-userinfo-bpm-name-display", 1, "adf-userinfo-name"], ["id", "identity-user-image"], [4, "ngIf", "ngIfElse"], ["initialTemplate", ""], [1, "adf-userinfo-profile-container"], ["id", "logged-user-img", "alt", "user-info-profile-button", 1, "adf-userinfo-profile-image", 3, "src"], [3, "innerHTML"], ["id", "ecm-user-image", 4, "ngIf", "ngIfElse"], ["showBpmUserImage", ""], ["id", "ecm-user-image"], ["class", "adf-userinfo-profile-container", 4, "ngIf", "ngIfElse"], [3, "outerHTML"], ["id", "bpm-user-image", 4, "ngIf"], ["id", "bpm-user-image"], ["id", "ecm-panel", 3, "label"], ["class", "adf-userinfo-card", 4, "ngIf"], [1, "adf-userinfo-card"], [1, "adf-userinfo-card-header"], ["class", "adf-userinfo-profile-container adf-hide-small", 4, "ngIf", "ngIfElse"], ["id", "ecm-username", 1, "adf-userinfo-title"], [1, "adf-userinfo-supporting-text"], [1, "adf-userinfo-detail"], ["id", "ecm-full-name", 1, "adf-userinfo__detail-title"], ["id", "ecm-email", 1, "adf-userinfo__detail-profile"], ["id", "ecm-job-title-label", 1, "adf-userinfo__secondary-info"], ["id", "ecm-job-title", 1, "adf-userinfo__detail-profile"], [1, "adf-userinfo-profile-container", "adf-hide-small"], ["id", "ecm-user-detail-image", "alt", "ecm-profile-image", 1, "adf-userinfo-profile-picture", 3, "src"], ["id", "bpm-panel", 3, "label"], ["class", "adf-userinfo-profile-picture adf-hide-small", "id", "bpm-user-detail-image", "alt", "bpm-profile-image", 3, "src", 4, "ngIf", "ngIfElse"], ["id", "bpm-username", 1, "adf-userinfo-title"], ["id", "bpm-full-name", 1, "adf-userinfo__detail-title"], ["id", "bpm-email", 1, "adf-userinfo__detail-profile"], ["id", "bpm-tenant", 1, "adf-userinfo__secondary-info"], [1, "adf-userinfo__detail-profile"], ["id", "bpm-user-detail-image", "alt", "bpm-profile-image", 1, "adf-userinfo-profile-picture", "adf-hide-small", 3, "src"], ["id", "identity-panel"], ["id", "identity-username", 1, "adf-userinfo-title"], ["id", "identity-full-name", 1, "adf-userinfo__detail-title"], ["id", "identity-email", 1, "adf-userinfo__detail-profile"]], template: function UserInfoComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, UserInfoComponent_div_0_Template, 16, 20, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.isLoggedIn);
    } }, directives: [ɵngcc5.NgIf, ɵngcc6.MatButton, ɵngcc7.MatMenuTrigger, ɵngcc7._MatMenu, ɵngcc8.MatTabGroup, ɵngcc8.MatTab, ɵngcc9.MatCard, ɵngcc9.MatCardHeader, ɵngcc9.MatCardContent], pipes: [ɵngcc5.AsyncPipe, ɵngcc10.FullNamePipe, ɵngcc11.InitialUsernamePipe, ɵngcc12.TranslatePipe], encapsulation: 2 });
UserInfoComponent.ctorParameters = () => [
    { type: EcmUserService },
    { type: BpmUserService },
    { type: IdentityUserService },
    { type: AuthenticationService }
];
UserInfoComponent.propDecorators = {
    trigger: [{ type: ViewChild, args: [MatMenuTrigger,] }],
    ecmBackgroundImage: [{ type: Input }],
    bpmBackgroundImage: [{ type: Input }],
    menuPositionX: [{ type: Input }],
    menuPositionY: [{ type: Input }],
    showName: [{ type: Input }],
    namePosition: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(UserInfoComponent, [{
        type: Component,
        args: [{
                selector: 'adf-userinfo',
                template: "<div id=\"userinfo_container\" [class.adf-userinfo-name-right]=\"showOnRight\" (keyup)=\"onKeyPress($event)\"\n     class=\"adf-userinfo-container\" *ngIf=\"isLoggedIn\">\n\n    <ng-container *ngIf=\"showName\">\n        <span *ngIf=\"identityUser$ | async as identityUser; else showBpmAndEcmUserFullNames\"\n              id=\"adf-userinfo-identity-name-display\"\n              class=\"adf-userinfo-name\">{{identityUser | fullName}}</span>\n        <ng-template #showBpmAndEcmUserFullNames>\n            <span *ngIf=\"ecmUser$ | async as ecmUser; else showBpmUserFullName\" id=\"adf-userinfo-ecm-name-display\"\n                  class=\"adf-userinfo-name\">{{ecmUser | fullName}}</span>\n            <ng-template #showBpmUserFullName>\n                <span *ngIf=\"bpmUser$ | async as bpmUser\" id=\"adf-userinfo-bpm-name-display\"\n                      class=\"adf-userinfo-name\">{{bpmUser | fullName}}</span>\n            </ng-template>\n        </ng-template>\n    </ng-container>\n\n    <button mat-button [matMenuTriggerFor]=\"menu\" class=\"adf-userinfo-menu_button\"\n            data-automation-id=\"adf-user-profile\">\n        <div class=\"adf-userinfo-button-profile\" id=\"user-profile\">\n            <div *ngIf=\"identityUser$ | async as identityUser; else showBpmAndEcmUserImage\" id=\"identity-user-image\">\n                <div *ngIf=\"(ecmUser$ | async)?.avatarId as avatarId; else initialTemplate\">\n                    <div class=\"adf-userinfo-profile-container\">\n                        <img id=\"logged-user-img\" [src]=\"getEcmAvatar(avatarId)\" alt=\"user-info-profile-button\"\n                             class=\"adf-userinfo-profile-image\"/>\n                    </div>\n                </div>\n                <ng-template #initialTemplate>\n                    <div [innerHTML]=\"identityUser | usernameInitials:'adf-userinfo-pic'\"></div>\n                </ng-template>\n            </div>\n            <ng-template #showBpmAndEcmUserImage>\n                <div *ngIf=\"ecmUser$ | async as ecmUser; else showBpmUserImage\" id=\"ecm-user-image\">\n                    <div *ngIf=\"ecmUser.avatarId; else initialTemplate\" class=\"adf-userinfo-profile-container\">\n                        <img id=\"logged-user-img\" [src]=\"getEcmAvatar(ecmUser.avatarId)\" alt=\"user-info-profile-button\"\n                             class=\"adf-userinfo-profile-image\"/>\n                    </div>\n                    <ng-template #initialTemplate>\n                        <div [outerHTML]=\"ecmUser | usernameInitials:'adf-userinfo-pic'\"></div>\n                    </ng-template>\n                </div>\n                <ng-template #showBpmUserImage>\n                    <div *ngIf=\"bpmUser$ | async as bpmUser\" id=\"bpm-user-image\">\n                        <div *ngIf=\"bpmUser.pictureId; else initialTemplate\" class=\"adf-userinfo-profile-container\">\n                            <img id=\"logged-user-img\" [src]=\"getBpmUserImage()\" alt=\"user-info-profile-button\"\n                                 class=\"adf-userinfo-profile-image\"/>\n                        </div>\n                        <ng-template #initialTemplate>\n                            <div [outerHTML]=\"bpmUser | usernameInitials:'adf-userinfo-pic'\"></div>\n                        </ng-template>\n                    </div>\n                </ng-template>\n            </ng-template>\n        </div>\n    </button>\n    <mat-menu #menu=\"matMenu\" id=\"user-profile-lists\" [xPosition]=\"menuPositionX\" [yPosition]=\"menuPositionY\"\n              [overlapTrigger]=\"false\" class=\"adf-userinfo-menu\">\n        <mat-tab-group id=\"tab-group-env\" (click)=\"stopClosing($event)\" selectedIndex=\"0\"\n                       class=\"adf-userinfo-tab\" [class.adf-hide-tab]=\"!(bpmUser$ | async) || !(ecmUser$ | async)\">\n            <mat-tab id=\"ecm-panel\" label=\"{{ 'USER_PROFILE.TAB.CS' | translate }}\"\n                     *ngIf=\"mode==='CONTENT' || mode==='ALL'\">\n                <mat-card class=\"adf-userinfo-card\" *ngIf=\"ecmUser$ | async as ecmUser\">\n                    <mat-card-header class=\"adf-userinfo-card-header\"\n                                     [style.background-image]=\"'url(' + ecmBackgroundImage + ')'\">\n                        <div *ngIf=\"ecmUser.avatarId; else initialTemplate\"\n                             class=\"adf-userinfo-profile-container adf-hide-small\">\n                            <img class=\"adf-userinfo-profile-picture\" id=\"ecm-user-detail-image\"\n                                 alt=\"ecm-profile-image\" [src]=\"getEcmAvatar(ecmUser.avatarId)\"/>\n                        </div>\n                        <ng-template #initialTemplate>\n                            <div\n                                [outerHTML]=\"ecmUser | usernameInitials:'adf-userinfo-profile-initials adf-hide-small'\"></div>\n                        </ng-template>\n\n                        <div class=\"adf-userinfo-title\" id=\"ecm-username\">{{ecmUser | fullName}}</div>\n                    </mat-card-header>\n                    <mat-card-content>\n                        <div class=\"adf-userinfo-supporting-text\">\n                            <div class=\"adf-userinfo-detail\">\n                                <span id=\"ecm-full-name\"\n                                      class=\"adf-userinfo__detail-title\">{{ecmUser | fullName}}</span>\n                                <span class=\"adf-userinfo__detail-profile\" id=\"ecm-email\"> {{ecmUser.email}} </span>\n                            </div>\n                            <div class=\"adf-userinfo-detail\">\n                                    <span class=\"adf-userinfo__secondary-info\" id=\"ecm-job-title-label\">\n                                        {{ 'USER_PROFILE.LABELS.ECM.JOB_TITLE' | translate }}\n                                        <span id=\"ecm-job-title\"\n                                              class=\"adf-userinfo__detail-profile\"> {{ ecmUser.jobTitle ? ecmUser.jobTitle : 'N/A' }} </span>\n                                    </span>\n                            </div>\n                        </div>\n                    </mat-card-content>\n                </mat-card>\n            </mat-tab>\n            <mat-tab id=\"bpm-panel\" label=\"{{ 'USER_PROFILE.TAB.PS' | translate }}\"\n                     *ngIf=\"mode==='PROCESS' || mode==='ALL'\">\n                <mat-card class=\"adf-userinfo-card\" *ngIf=\"bpmUser$ | async as bpmUser\">\n                    <mat-card-header class=\"adf-userinfo-card-header\"\n                                     [style.background-image]=\"'url(' + bpmBackgroundImage + ')'\">\n                        <img *ngIf=\"bpmUser.pictureId; else initialTemplate\"\n                             class=\"adf-userinfo-profile-picture adf-hide-small\" id=\"bpm-user-detail-image\"\n                             alt=\"bpm-profile-image\" [src]=\"getBpmUserImage()\"/>\n                        <ng-template #initialTemplate>\n                            <div\n                                [outerHTML]=\"bpmUser | usernameInitials:'adf-userinfo-profile-initials adf-hide-small'\"></div>\n                        </ng-template>\n                        <div class=\"adf-userinfo-title\" id=\"bpm-username\">{{bpmUser | fullName}}</div>\n                    </mat-card-header>\n                    <mat-card-content>\n                        <div class=\"adf-userinfo-supporting-text\">\n                            <div class=\"adf-userinfo-detail\">\n                                <span id=\"bpm-full-name\"\n                                      class=\"adf-userinfo__detail-title\">{{ bpmUser | fullName }}</span>\n                                <span class=\"adf-userinfo__detail-profile\" id=\"bpm-email\"> {{bpmUser.email}} </span>\n                            </div>\n                            <div class=\"adf-userinfo-detail\">\n                                <span id=\"bpm-tenant\" class=\"adf-userinfo__secondary-info\">\n                                    {{ 'USER_PROFILE.LABELS.BPM.TENANT' | translate }}\n                                    <span\n                                        class=\"adf-userinfo__detail-profile\">{{ bpmUser.tenantName ? bpmUser.tenantName : '' }}</span>\n                                </span>\n                            </div>\n                        </div>\n                    </mat-card-content>\n                </mat-card>\n            </mat-tab>\n            <mat-tab id=\"identity-panel\" *ngIf=\"mode==='SSO'\">\n                <mat-card class=\"adf-userinfo-card\" *ngIf=\"identityUser$ | async as identityUser\">\n                    <mat-card-header class=\"adf-userinfo-card-header\"\n                                     [style.background-image]=\"'url(' + bpmBackgroundImage + ')'\">\n                        <div *ngIf=\"ecmUser$ | async as ecmUser\">\n                            <div *ngIf=\"ecmUser.avatarId; else initialTemplate\"\n                                 class=\"adf-userinfo-profile-container adf-hide-small\">\n                                <img class=\"adf-userinfo-profile-picture\" id=\"ecm-user-detail-image\"\n                                     alt=\"ecm-profile-image\" [src]=\"getEcmAvatar(ecmUser.avatarId)\"/>\n                            </div>\n                        </div>\n                        <ng-template #initialTemplate>\n                            <div\n                                [outerHTML]=\"identityUser | usernameInitials:'adf-userinfo-profile-initials adf-hide-small'\"></div>\n                        </ng-template>\n                        <div class=\"adf-userinfo-title\" id=\"identity-username\">{{identityUser | fullName}}</div>\n                    </mat-card-header>\n                    <mat-card-content>\n                        <div class=\"adf-userinfo-supporting-text\">\n                            <div class=\"adf-userinfo-detail\">\n                                <span id=\"identity-full-name\"\n                                      class=\"adf-userinfo__detail-title\">{{identityUser | fullName}}</span>\n                                <span class=\"adf-userinfo__detail-profile\"\n                                      id=\"identity-email\"> {{identityUser.email}} </span>\n                            </div>\n                        </div>\n                    </mat-card-content>\n                </mat-card>\n            </mat-tab>\n        </mat-tab-group>\n    </mat-menu>\n</div>\n",
                encapsulation: ViewEncapsulation.None
            }]
    }], function () { return [{ type: ɵngcc1.EcmUserService }, { type: ɵngcc2.BpmUserService }, { type: ɵngcc3.IdentityUserService }, { type: ɵngcc4.AuthenticationService }]; }, { ecmBackgroundImage: [{
            type: Input
        }], bpmBackgroundImage: [{
            type: Input
        }], menuPositionX: [{
            type: Input
        }], menuPositionY: [{
            type: Input
        }], showName: [{
            type: Input
        }], namePosition: [{
            type: Input
        }], trigger: [{
            type: ViewChild,
            args: [MatMenuTrigger]
        }] }); })();
