// @alfresco/adf-core/esm2015/info-drawer/info-drawer-layout.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, Directive, Input, ViewEncapsulation } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function InfoDrawerLayoutComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 2);
    ɵngcc0.ɵɵelementStart(1, "div", 3);
    ɵngcc0.ɵɵprojection(2, 1);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "div", 4);
    ɵngcc0.ɵɵprojection(4, 2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} }
const _c0 = [[["", "info-drawer-content", ""]], [["", "info-drawer-title", ""]], [["", "info-drawer-buttons", ""]]];
const _c1 = ["[info-drawer-content]", "[info-drawer-title]", "[info-drawer-buttons]"];
export class InfoDrawerLayoutComponent {
    constructor() {
        this.showHeader = true;
    }
}
InfoDrawerLayoutComponent.ɵfac = function InfoDrawerLayoutComponent_Factory(t) { return new (t || InfoDrawerLayoutComponent)(); };
InfoDrawerLayoutComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: InfoDrawerLayoutComponent, selectors: [["adf-info-drawer-layout"]], hostAttrs: [1, "adf-info-drawer-layout"], inputs: { showHeader: "showHeader" }, ngContentSelectors: _c1, decls: 3, vars: 1, consts: [["class", "adf-info-drawer-layout-header", 4, "ngIf"], [1, "adf-info-drawer-layout-content"], [1, "adf-info-drawer-layout-header"], [1, "adf-info-drawer-layout-header-title"], [1, "adf-info-drawer-layout-header-buttons"]], template: function InfoDrawerLayoutComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c0);
        ɵngcc0.ɵɵtemplate(0, InfoDrawerLayoutComponent_div_0_Template, 5, 0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵprojection(2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.showHeader);
    } }, directives: [ɵngcc1.NgIf], encapsulation: 2 });
InfoDrawerLayoutComponent.propDecorators = {
    showHeader: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InfoDrawerLayoutComponent, [{
        type: Component,
        args: [{
                selector: 'adf-info-drawer-layout',
                template: "<div *ngIf=\"showHeader\" class=\"adf-info-drawer-layout-header\">\n    <div class=\"adf-info-drawer-layout-header-title\">\n        <ng-content select=\"[info-drawer-title]\"></ng-content>\n    </div>\n    <div class=\"adf-info-drawer-layout-header-buttons\">\n        <ng-content select=\"[info-drawer-buttons]\"></ng-content>\n    </div>\n</div>\n<div class=\"adf-info-drawer-layout-content\">\n    <ng-content select=\"[info-drawer-content]\"></ng-content>\n</div>\n",
                encapsulation: ViewEncapsulation.None,
                host: { 'class': 'adf-info-drawer-layout' }
            }]
    }], function () { return []; }, { showHeader: [{
            type: Input
        }] }); })();
export class InfoDrawerTitleDirective {
}
InfoDrawerTitleDirective.ɵfac = function InfoDrawerTitleDirective_Factory(t) { return new (t || InfoDrawerTitleDirective)(); };
InfoDrawerTitleDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: InfoDrawerTitleDirective, selectors: [["", "adf-info-drawer-title", ""], ["", "info-drawer-title", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InfoDrawerTitleDirective, [{
        type: Directive,
        args: [{ selector: '[adf-info-drawer-title], [info-drawer-title]' }]
    }], null, null); })();
export class InfoDrawerButtonsDirective {
}
InfoDrawerButtonsDirective.ɵfac = function InfoDrawerButtonsDirective_Factory(t) { return new (t || InfoDrawerButtonsDirective)(); };
InfoDrawerButtonsDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: InfoDrawerButtonsDirective, selectors: [["", "adf-info-drawer-buttons", ""], ["", "info-drawer-buttons", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InfoDrawerButtonsDirective, [{
        type: Directive,
        args: [{ selector: '[adf-info-drawer-buttons], [info-drawer-buttons]' }]
    }], null, null); })();
export class InfoDrawerContentDirective {
}
InfoDrawerContentDirective.ɵfac = function InfoDrawerContentDirective_Factory(t) { return new (t || InfoDrawerContentDirective)(); };
InfoDrawerContentDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: InfoDrawerContentDirective, selectors: [["", "adf-info-drawer-content", ""], ["", "info-drawer-content", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InfoDrawerContentDirective, [{
        type: Directive,
        args: [{ selector: '[adf-info-drawer-content], [info-drawer-content]' }]
    }], null, null); })();
