// @alfresco/adf-core/esm2015/viewer/components/pdf-viewer.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { __awaiter } from "tslib";
import { Component, TemplateRef, HostListener, Output, Input, ViewEncapsulation, EventEmitter } from '@angular/core';
import { MatDialog } from '@angular/material/dialog';
import { LogService } from '../../services/log.service';
import { RenderingQueueServices } from '../services/rendering-queue.services';
import { PdfPasswordDialogComponent } from './pdf-viewer-password-dialog';
import { AppConfigService } from './../../app-config/app-config.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/material/dialog';
import * as ɵngcc2 from '../services/rendering-queue.services';
import * as ɵngcc3 from '../../services/log.service';
import * as ɵngcc4 from './../../app-config/app-config.service';
import * as ɵngcc5 from '@angular/common';
import * as ɵngcc6 from '@angular/material/progress-bar';
import * as ɵngcc7 from '@angular/material/button';
import * as ɵngcc8 from '@angular/material/icon';
import * as ɵngcc9 from './pdf-viewer-thumbnails.component';
import * as ɵngcc10 from '../../toolbar/toolbar.component';
import * as ɵngcc11 from '../../toolbar/toolbar-divider.component';
import * as ɵngcc12 from '@ngx-translate/core';

function PdfViewerComponent_ng_container_1_ng_container_9_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainer(0);
} }
function PdfViewerComponent_ng_container_1_ng_container_9_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, PdfViewerComponent_ng_container_1_ng_container_9_ng_container_1_Template, 1, 0, "ng-container", 13);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r2.thumbnailsTemplate)("ngTemplateOutletContext", ctx_r2.pdfThumbnailsContext);
} }
function PdfViewerComponent_ng_container_1_ng_container_10_Template(rf, ctx) { if (rf & 1) {
    const _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "adf-pdf-thumbnails", 14);
    ɵngcc0.ɵɵlistener("close", function PdfViewerComponent_ng_container_1_ng_container_10_Template_adf_pdf_thumbnails_close_1_listener() { ɵngcc0.ɵɵrestoreView(_r6); const ctx_r5 = ɵngcc0.ɵɵnextContext(2); return ctx_r5.toggleThumbnails(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r3 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("pdfViewer", ctx_r3.pdfViewer);
} }
function PdfViewerComponent_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    const _r8 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "div", 9);
    ɵngcc0.ɵɵelementStart(2, "div", 10);
    ɵngcc0.ɵɵelementStart(3, "div", 11);
    ɵngcc0.ɵɵelementStart(4, "button", 12);
    ɵngcc0.ɵɵlistener("click", function PdfViewerComponent_ng_container_1_Template_button_click_4_listener() { ɵngcc0.ɵɵrestoreView(_r8); const ctx_r7 = ɵngcc0.ɵɵnextContext(); return ctx_r7.toggleThumbnails(); });
    ɵngcc0.ɵɵpipe(5, "translate");
    ɵngcc0.ɵɵpipe(6, "translate");
    ɵngcc0.ɵɵelementStart(7, "mat-icon");
    ɵngcc0.ɵɵtext(8, "close");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(9, PdfViewerComponent_ng_container_1_ng_container_9_Template, 2, 2, "ng-container", 1);
    ɵngcc0.ɵɵtemplate(10, PdfViewerComponent_ng_container_1_ng_container_10_Template, 2, 1, "ng-container", 1);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵpropertyInterpolate("title", ɵngcc0.ɵɵpipeBind1(6, 6, "ADF_VIEWER.ACTIONS.CLOSE"));
    ɵngcc0.ɵɵattribute("aria-label", ɵngcc0.ɵɵpipeBind1(5, 4, "ADF_VIEWER.ARIA.THUMBNAILS_PANLEL_CLOSE"));
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.thumbnailsTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.thumbnailsTemplate);
} }
function PdfViewerComponent_div_8_ng_container_2_Template(rf, ctx) { if (rf & 1) {
    const _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "button", 25);
    ɵngcc0.ɵɵlistener("click", function PdfViewerComponent_div_8_ng_container_2_Template_button_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r12); const ctx_r11 = ɵngcc0.ɵɵnextContext(2); return ctx_r11.toggleThumbnails(); });
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementStart(3, "mat-icon");
    ɵngcc0.ɵɵtext(4, "dashboard");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(5, "adf-toolbar-divider");
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r9 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("disabled", ctx_r9.isPanelDisabled);
    ɵngcc0.ɵɵattribute("aria-label", ɵngcc0.ɵɵpipeBind1(2, 3, "ADF_VIEWER.ARIA.THUMBNAILS"))("aria-expanded", ctx_r9.showThumbnails);
} }
function PdfViewerComponent_div_8_Template(rf, ctx) { if (rf & 1) {
    const _r14 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 15);
    ɵngcc0.ɵɵelementStart(1, "adf-toolbar");
    ɵngcc0.ɵɵtemplate(2, PdfViewerComponent_div_8_ng_container_2_Template, 6, 5, "ng-container", 1);
    ɵngcc0.ɵɵelementStart(3, "button", 16);
    ɵngcc0.ɵɵlistener("click", function PdfViewerComponent_div_8_Template_button_click_3_listener() { ɵngcc0.ɵɵrestoreView(_r14); const ctx_r13 = ɵngcc0.ɵɵnextContext(); return ctx_r13.previousPage(); });
    ɵngcc0.ɵɵpipe(4, "translate");
    ɵngcc0.ɵɵpipe(5, "translate");
    ɵngcc0.ɵɵelementStart(6, "mat-icon");
    ɵngcc0.ɵɵtext(7, "keyboard_arrow_up");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(8, "button", 17);
    ɵngcc0.ɵɵlistener("click", function PdfViewerComponent_div_8_Template_button_click_8_listener() { ɵngcc0.ɵɵrestoreView(_r14); const ctx_r15 = ɵngcc0.ɵɵnextContext(); return ctx_r15.nextPage(); });
    ɵngcc0.ɵɵpipe(9, "translate");
    ɵngcc0.ɵɵpipe(10, "translate");
    ɵngcc0.ɵɵelementStart(11, "mat-icon");
    ɵngcc0.ɵɵtext(12, "keyboard_arrow_down");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(13, "div", 18);
    ɵngcc0.ɵɵelementStart(14, "span");
    ɵngcc0.ɵɵtext(15);
    ɵngcc0.ɵɵpipe(16, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(17, "input", 19, 20);
    ɵngcc0.ɵɵlistener("keyup.enter", function PdfViewerComponent_div_8_Template_input_keyup_enter_17_listener() { ɵngcc0.ɵɵrestoreView(_r14); const _r10 = ɵngcc0.ɵɵreference(18); const ctx_r16 = ɵngcc0.ɵɵnextContext(); return ctx_r16.inputPage(_r10.value); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(19, "span");
    ɵngcc0.ɵɵtext(20);
    ɵngcc0.ɵɵpipe(21, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(22, "div", 21);
    ɵngcc0.ɵɵtext(23);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(24, "button", 22);
    ɵngcc0.ɵɵlistener("click", function PdfViewerComponent_div_8_Template_button_click_24_listener() { ɵngcc0.ɵɵrestoreView(_r14); const ctx_r17 = ɵngcc0.ɵɵnextContext(); return ctx_r17.zoomIn(); });
    ɵngcc0.ɵɵpipe(25, "translate");
    ɵngcc0.ɵɵpipe(26, "translate");
    ɵngcc0.ɵɵelementStart(27, "mat-icon");
    ɵngcc0.ɵɵtext(28, "zoom_in");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(29, "button", 23);
    ɵngcc0.ɵɵlistener("click", function PdfViewerComponent_div_8_Template_button_click_29_listener() { ɵngcc0.ɵɵrestoreView(_r14); const ctx_r18 = ɵngcc0.ɵɵnextContext(); return ctx_r18.zoomOut(); });
    ɵngcc0.ɵɵpipe(30, "translate");
    ɵngcc0.ɵɵpipe(31, "translate");
    ɵngcc0.ɵɵelementStart(32, "mat-icon");
    ɵngcc0.ɵɵtext(33, "zoom_out");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(34, "button", 24);
    ɵngcc0.ɵɵlistener("click", function PdfViewerComponent_div_8_Template_button_click_34_listener() { ɵngcc0.ɵɵrestoreView(_r14); const ctx_r19 = ɵngcc0.ɵɵnextContext(); return ctx_r19.pageFit(); });
    ɵngcc0.ɵɵpipe(35, "translate");
    ɵngcc0.ɵɵpipe(36, "translate");
    ɵngcc0.ɵɵelementStart(37, "mat-icon");
    ɵngcc0.ɵɵtext(38, "zoom_out_map");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1.allowThumbnails);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate("title", ɵngcc0.ɵɵpipeBind1(4, 16, "ADF_VIEWER.ARIA.PREVIOUS_PAGE"));
    ɵngcc0.ɵɵattribute("aria-label", ɵngcc0.ɵɵpipeBind1(5, 18, "ADF_VIEWER.ARIA.PREVIOUS_PAGE"));
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵpropertyInterpolate("title", ɵngcc0.ɵɵpipeBind1(9, 20, "ADF_VIEWER.ARIA.NEXT_PAGE"));
    ɵngcc0.ɵɵattribute("aria-label", ɵngcc0.ɵɵpipeBind1(10, 22, "ADF_VIEWER.ARIA.NEXT_PAGE"));
    ɵngcc0.ɵɵadvance(7);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind1(16, 24, "ADF_VIEWER.PAGE_LABEL.SHOWING"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵpropertyInterpolate("value", ctx_r1.displayPage);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate2("", ɵngcc0.ɵɵpipeBind1(21, 26, "ADF_VIEWER.PAGE_LABEL.OF"), " ", ctx_r1.totalPages, "");
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r1.currentScaleText, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate("title", ɵngcc0.ɵɵpipeBind1(25, 28, "ADF_VIEWER.ARIA.ZOOM_IN"));
    ɵngcc0.ɵɵattribute("aria-label", ɵngcc0.ɵɵpipeBind1(26, 30, "ADF_VIEWER.ARIA.ZOOM_IN"));
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵpropertyInterpolate("title", ɵngcc0.ɵɵpipeBind1(30, 32, "ADF_VIEWER.ARIA.ZOOM_OUT"));
    ɵngcc0.ɵɵattribute("aria-label", ɵngcc0.ɵɵpipeBind1(31, 34, "ADF_VIEWER.ARIA.ZOOM_OUT"));
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵpropertyInterpolate("title", ɵngcc0.ɵɵpipeBind1(35, 36, "ADF_VIEWER.ARIA.FIT_PAGE"));
    ɵngcc0.ɵɵattribute("aria-label", ɵngcc0.ɵɵpipeBind1(36, 38, "ADF_VIEWER.ARIA.FIT_PAGE"));
} }
export class PdfViewerComponent {
    constructor(dialog, renderingQueueServices, logService, appConfigService) {
        this.dialog = dialog;
        this.renderingQueueServices = renderingQueueServices;
        this.logService = logService;
        this.appConfigService = appConfigService;
        this.showToolbar = true;
        this.allowThumbnails = false;
        this.thumbnailsTemplate = null;
        this.cacheType = '';
        this.rendered = new EventEmitter();
        this.error = new EventEmitter();
        this.close = new EventEmitter();
        this.currentScaleMode = 'auto';
        this.currentScale = 1;
        this.MAX_AUTO_SCALE = 1.25;
        this.DEFAULT_SCALE_DELTA = 1.1;
        this.MIN_SCALE = 0.25;
        this.MAX_SCALE = 10.0;
        this.isPanelDisabled = true;
        this.showThumbnails = false;
        this.pdfThumbnailsContext = { viewer: null };
        this.eventBus = new pdfjsViewer.EventBus();
        this.pdfjsDefaultOptions = {
            disableAutoFetch: true,
            disableStream: true
        };
        this.onPageChange = this.onPageChange.bind(this);
        this.onPagesLoaded = this.onPagesLoaded.bind(this);
        this.onPageRendered = this.onPageRendered.bind(this);
        this.randomPdfId = this.generateUuid();
        this.currentScale = this.getUserScaling();
    }
    get currentScaleText() {
        return Math.round(this.currentScale * 100) + '%';
    }
    getUserScaling() {
        const scaleConfig = this.appConfigService.get('adf-viewer.pdf-viewer-scaling', undefined) / 100;
        if (scaleConfig) {
            return this.checkLimits(scaleConfig);
        }
        else {
            return 1;
        }
    }
    checkLimits(scaleConfig) {
        if (scaleConfig > this.MAX_SCALE) {
            return this.MAX_SCALE;
        }
        else if (scaleConfig < this.MIN_SCALE) {
            return this.MIN_SCALE;
        }
        else {
            return scaleConfig;
        }
    }
    ngOnChanges(changes) {
        const blobFile = changes['blobFile'];
        if (blobFile && blobFile.currentValue) {
            const reader = new FileReader();
            reader.onload = () => __awaiter(this, void 0, void 0, function* () {
                const pdfSource = Object.assign(Object.assign({}, this.pdfjsDefaultOptions), { data: reader.result, withCredentials: this.appConfigService.get('auth.withCredentials', undefined) });
                this.executePdf(pdfSource);
            });
            reader.readAsArrayBuffer(blobFile.currentValue);
        }
        const urlFile = changes['urlFile'];
        if (urlFile && urlFile.currentValue) {
            const pdfSource = Object.assign(Object.assign({}, this.pdfjsDefaultOptions), { url: urlFile.currentValue, withCredentials: this.appConfigService.get('auth.withCredentials', undefined) });
            if (this.cacheType) {
                pdfSource.httpHeaders = {
                    'Cache-Control': this.cacheType
                };
            }
            this.executePdf(pdfSource);
        }
        if (!this.urlFile && !this.blobFile) {
            throw new Error('Attribute urlFile or blobFile is required');
        }
    }
    executePdf(pdfOptions) {
        pdfjsLib.GlobalWorkerOptions.workerSrc = 'pdf.worker.min.js';
        this.loadingTask = pdfjsLib.getDocument(pdfOptions);
        this.loadingTask.onPassword = (callback, reason) => {
            this.onPdfPassword(callback, reason);
        };
        this.loadingTask.onProgress = (progressData) => {
            const level = progressData.loaded / progressData.total;
            this.loadingPercent = Math.round(level * 100);
        };
        this.loadingTask.promise.then((pdfDocument) => {
            this.totalPages = pdfDocument.numPages;
            this.page = 1;
            this.displayPage = 1;
            this.initPDFViewer(pdfDocument);
            return pdfDocument.getPage(1);
        })
            .then(() => this.scalePage('auto'))
            .catch(() => this.error.emit());
    }
    initPDFViewer(pdfDocument) {
        const viewer = this.getViewer();
        const container = this.getDocumentContainer();
        if (viewer && container) {
            this.pdfViewer = new pdfjsViewer.PDFViewer({
                container: container,
                viewer: viewer,
                renderingQueue: this.renderingQueueServices,
                eventBus: this.eventBus
            });
            this.eventBus.on('pagechanging', this.onPageChange);
            this.eventBus.on('pagesloaded', this.onPagesLoaded);
            this.eventBus.on('textlayerrendered', this.onPageRendered);
            this.renderingQueueServices.setViewer(this.pdfViewer);
            this.pdfViewer.setDocument(pdfDocument);
            this.pdfThumbnailsContext.viewer = this.pdfViewer;
        }
    }
    ngOnDestroy() {
        if (this.pdfViewer) {
            this.eventBus.off('pagechanging');
            this.eventBus.off('pagesloaded');
            this.eventBus.off('textlayerrendered');
        }
        if (this.loadingTask) {
            try {
                this.loadingTask.destroy();
            }
            catch (_a) {
            }
            this.loadingTask = null;
        }
    }
    toggleThumbnails() {
        this.showThumbnails = !this.showThumbnails;
    }
    scalePage(scaleMode) {
        this.currentScaleMode = scaleMode;
        const viewerContainer = document.getElementById(`${this.randomPdfId}-viewer-main-container`);
        const documentContainer = this.getDocumentContainer();
        if (this.pdfViewer && documentContainer) {
            let widthContainer;
            let heightContainer;
            if (viewerContainer && viewerContainer.clientWidth <= documentContainer.clientWidth) {
                widthContainer = viewerContainer.clientWidth;
                heightContainer = viewerContainer.clientHeight;
            }
            else {
                widthContainer = documentContainer.clientWidth;
                heightContainer = documentContainer.clientHeight;
            }
            const currentPage = this.pdfViewer._pages[this.pdfViewer._currentPageNumber - 1];
            const padding = 20;
            const pageWidthScale = (widthContainer - padding) / currentPage.width * currentPage.scale;
            const pageHeightScale = (heightContainer - padding) / currentPage.width * currentPage.scale;
            let scale = this.getUserScaling();
            if (!scale) {
                switch (this.currentScaleMode) {
                    case 'page-actual':
                        scale = 1;
                        break;
                    case 'page-width':
                        scale = pageWidthScale;
                        break;
                    case 'page-height':
                        scale = pageHeightScale;
                        break;
                    case 'page-fit':
                        scale = Math.min(pageWidthScale, pageHeightScale);
                        break;
                    case 'auto':
                        let horizontalScale;
                        if (this.isLandscape) {
                            horizontalScale = Math.min(pageHeightScale, pageWidthScale);
                        }
                        else {
                            horizontalScale = pageWidthScale;
                        }
                        horizontalScale = Math.round(horizontalScale);
                        scale = Math.min(this.MAX_AUTO_SCALE, horizontalScale);
                        scale = this.checkPageFitInContainer(scale);
                        break;
                    default:
                        this.logService.error('pdfViewSetScale: \'' + scaleMode + '\' is an unknown zoom value.');
                        return;
                }
                this.setScaleUpdatePages(scale);
            }
            else {
                this.currentScale = 0;
                scale = this.checkPageFitInContainer(scale);
                this.setScaleUpdatePages(scale);
            }
        }
    }
    getDocumentContainer() {
        return document.getElementById(`${this.randomPdfId}-viewer-pdf-viewer`);
    }
    getViewer() {
        return document.getElementById(`${this.randomPdfId}-viewer-viewerPdf`);
    }
    checkPageFitInContainer(scale) {
        const documentContainerSize = this.getDocumentContainer();
        const page = this.pdfViewer._pages[this.pdfViewer._currentPageNumber - 1];
        if (page.width > documentContainerSize.clientWidth) {
            scale = Math.fround((documentContainerSize.clientWidth - 20) / page.width);
            if (scale < this.MIN_SCALE) {
                scale = this.MIN_SCALE;
            }
        }
        return scale;
    }
    setScaleUpdatePages(newScale) {
        if (this.pdfViewer) {
            if (!this.isSameScale(this.currentScale, newScale)) {
                this.currentScale = newScale;
                this.pdfViewer._pages.forEach(function (currentPage) {
                    currentPage.update(newScale);
                });
            }
            this.pdfViewer.update();
        }
    }
    isSameScale(oldScale, newScale) {
        return (newScale === oldScale);
    }
    isLandscape(width, height) {
        return (width > height);
    }
    onResize() {
        this.scalePage(this.currentScaleMode);
    }
    pageFit() {
        if (this.currentScaleMode !== 'page-fit') {
            this.scalePage('page-fit');
        }
        else {
            this.scalePage('auto');
        }
    }
    zoomIn(ticks) {
        let newScale = this.currentScale;
        do {
            newScale = (newScale * this.DEFAULT_SCALE_DELTA).toFixed(2);
            newScale = Math.ceil(newScale * 10) / 10;
            newScale = Math.min(this.MAX_SCALE, newScale);
        } while (--ticks > 0 && newScale < this.MAX_SCALE);
        this.currentScaleMode = 'auto';
        this.setScaleUpdatePages(newScale);
    }
    zoomOut(ticks) {
        let newScale = this.currentScale;
        do {
            newScale = (newScale / this.DEFAULT_SCALE_DELTA).toFixed(2);
            newScale = Math.floor(newScale * 10) / 10;
            newScale = Math.max(this.MIN_SCALE, newScale);
        } while (--ticks > 0 && newScale > this.MIN_SCALE);
        this.currentScaleMode = 'auto';
        this.setScaleUpdatePages(newScale);
    }
    previousPage() {
        if (this.pdfViewer && this.page > 1) {
            this.page--;
            this.displayPage = this.page;
            this.pdfViewer.currentPageNumber = this.page;
        }
    }
    nextPage() {
        if (this.pdfViewer && this.page < this.totalPages) {
            this.page++;
            this.displayPage = this.page;
            this.pdfViewer.currentPageNumber = this.page;
        }
    }
    inputPage(page) {
        const pageInput = parseInt(page, 10);
        if (!isNaN(pageInput) && pageInput > 0 && pageInput <= this.totalPages) {
            this.page = pageInput;
            this.displayPage = this.page;
            this.pdfViewer.currentPageNumber = this.page;
        }
        else {
            this.displayPage = this.page;
        }
    }
    onPageChange(event) {
        if (event.source && event.source.container.id === `${this.randomPdfId}-viewer-pdf-viewer`) {
            this.page = event.pageNumber;
            this.displayPage = event.pageNumber;
        }
    }
    onPdfPassword(callback, reason) {
        this.dialog
            .open(PdfPasswordDialogComponent, {
            width: '400px',
            data: { reason }
        })
            .afterClosed().subscribe((password) => {
            if (password) {
                callback(password);
            }
            else {
                this.close.emit();
            }
        });
    }
    onPageRendered() {
        this.rendered.emit();
    }
    onPagesLoaded() {
        this.isPanelDisabled = false;
    }
    handleKeyboardEvent(event) {
        const key = event.keyCode;
        if (key === 39) {
            this.nextPage();
        }
        else if (key === 37) {
            this.previousPage();
        }
    }
    generateUuid() {
        return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
            const r = Math.random() * 16 | 0, v = c === 'x' ? r : (r & 0x3 | 0x8);
            return v.toString(16);
        });
    }
}
PdfViewerComponent.ɵfac = function PdfViewerComponent_Factory(t) { return new (t || PdfViewerComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.MatDialog), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.RenderingQueueServices), ɵngcc0.ɵɵdirectiveInject(ɵngcc3.LogService), ɵngcc0.ɵɵdirectiveInject(ɵngcc4.AppConfigService)); };
PdfViewerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: PdfViewerComponent, selectors: [["adf-pdf-viewer"]], hostAttrs: [1, "adf-pdf-viewer"], hostBindings: function PdfViewerComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("keydown", function PdfViewerComponent_keydown_HostBindingHandler($event) { return ctx.handleKeyboardEvent($event); }, false, ɵngcc0.ɵɵresolveDocument);
    } }, inputs: { showToolbar: "showToolbar", allowThumbnails: "allowThumbnails", thumbnailsTemplate: "thumbnailsTemplate", cacheType: "cacheType", urlFile: "urlFile", blobFile: "blobFile", nameFile: "nameFile" }, outputs: { rendered: "rendered", error: "error", close: "close" }, features: [ɵngcc0.ɵɵProvidersFeature([RenderingQueueServices]), ɵngcc0.ɵɵNgOnChangesFeature], decls: 9, vars: 4, consts: [[1, "adf-pdf-viewer__container"], [4, "ngIf"], [1, "adf-pdf-viewer__content"], [1, "adf-viewer-pdf-viewer", 3, "id", "resize"], ["role", "document", "tabindex", "0", "aria-expanded", "true", 1, "adf-pdfViewer", 3, "id"], ["id", "loader-container", 1, "adf-loader-container"], [1, "adf-loader-item"], ["mode", "indeterminate"], ["class", "adf-pdf-viewer__toolbar", 4, "ngIf"], [1, "adf-pdf-viewer__thumbnails"], [1, "adf-thumbnails-template__container"], [1, "adf-thumbnails-template__buttons"], ["mat-icon-button", "", "data-automation-id", "adf-thumbnails-close", 3, "title", "click"], [4, "ngTemplateOutlet", "ngTemplateOutletContext"], [3, "pdfViewer", "close"], [1, "adf-pdf-viewer__toolbar"], ["id", "viewer-previous-page-button", "mat-icon-button", "", 3, "title", "click"], ["id", "viewer-next-page-button", "mat-icon-button", "", 3, "title", "click"], [1, "adf-pdf-viewer__toolbar-page-selector"], ["type", "text", "data-automation-id", "adf-page-selector", "pattern", "-?[0-9]*(\\.[0-9]+)?", 3, "value", "keyup.enter"], ["page", ""], ["data-automation-id", "adf-page-scale", 1, "adf-viewer__toolbar-page-scale"], ["id", "viewer-zoom-in-button", "mat-icon-button", "", 3, "title", "click"], ["id", "viewer-zoom-out-button", "mat-icon-button", "", 3, "title", "click"], ["id", "viewer-scale-page-button", "mat-icon-button", "", 3, "title", "click"], ["mat-icon-button", "", "data-automation-id", "adf-thumbnails-button", 3, "disabled", "click"]], template: function PdfViewerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, PdfViewerComponent_ng_container_1_Template, 11, 8, "ng-container", 1);
        ɵngcc0.ɵɵelementStart(2, "div", 2);
        ɵngcc0.ɵɵelementStart(3, "div", 3);
        ɵngcc0.ɵɵlistener("resize", function PdfViewerComponent_Template_div_resize_3_listener() { return ctx.onResize(); }, false, ɵngcc0.ɵɵresolveWindow);
        ɵngcc0.ɵɵelementStart(4, "div", 4);
        ɵngcc0.ɵɵelementStart(5, "div", 5);
        ɵngcc0.ɵɵelementStart(6, "div", 6);
        ɵngcc0.ɵɵelement(7, "mat-progress-bar", 7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(8, PdfViewerComponent_div_8_Template, 39, 40, "div", 8);
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showThumbnails);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("id", ctx.randomPdfId + "-viewer-pdf-viewer");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("id", ctx.randomPdfId + "-viewer-viewerPdf");
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showToolbar);
    } }, directives: [ɵngcc5.NgIf, ɵngcc6.MatProgressBar, ɵngcc7.MatButton, ɵngcc8.MatIcon, ɵngcc5.NgTemplateOutlet, ɵngcc9.PdfThumbListComponent, ɵngcc10.ToolbarComponent, ɵngcc11.ToolbarDividerComponent], pipes: [ɵngcc12.TranslatePipe], styles: [".adf-pdf-viewer .textLayer>span{color:transparent;cursor:text;position:absolute;transform-origin:0 0;white-space:pre}.adf-pdf-viewer .textLayer .highlight{background-color:#b400aa;border-radius:4px;margin:-1px;padding:1px}.adf-pdf-viewer .textLayer .highlight.begin{border-radius:4px 0 0 4px}.adf-pdf-viewer .textLayer .highlight.end{border-radius:0 4px 4px 0}.adf-pdf-viewer .textLayer .highlight.middle{border-radius:0}.adf-pdf-viewer .textLayer .highlight.selected{background-color:#006400}.adf-pdf-viewer .textLayer ::selection{background:#00f}.adf-pdf-viewer .textLayer .endOfContent{-webkit-user-select:none;bottom:0;cursor:default;display:block;left:0;position:absolute;right:0;top:100%;user-select:none;z-index:-1}.adf-pdf-viewer .textLayer .endOfContent.active{top:0}.adf-pdf-viewer .annotationLayer section{position:absolute}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.pushButton>a,.adf-pdf-viewer .annotationLayer .linkAnnotation>a{font-size:1em;height:100%;left:0;position:absolute;top:0;width:100%}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.pushButton>a:hover,.adf-pdf-viewer .annotationLayer .linkAnnotation>a:hover{background:#ff0;box-shadow:0 2px 10px #ff0;opacity:.2}.adf-pdf-viewer .annotationLayer .textAnnotation img{cursor:pointer;position:absolute}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input,.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.radioButton input,.adf-pdf-viewer .annotationLayer .choiceWidgetAnnotation select,.adf-pdf-viewer .annotationLayer .textWidgetAnnotation input,.adf-pdf-viewer .annotationLayer .textWidgetAnnotation textarea{background-color:rgba(0,54,255,.13);border:1px solid transparent;box-sizing:border-box;font-size:9px;height:100%;margin:0;padding:0 3px;vertical-align:top;width:100%}.adf-pdf-viewer .annotationLayer .choiceWidgetAnnotation select option{padding:0}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.radioButton input{border-radius:50%}.adf-pdf-viewer .annotationLayer .textWidgetAnnotation textarea{font:message-box;font-size:9px;resize:none}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input[disabled],.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.radioButton input[disabled],.adf-pdf-viewer .annotationLayer .choiceWidgetAnnotation select[disabled],.adf-pdf-viewer .annotationLayer .textWidgetAnnotation input[disabled],.adf-pdf-viewer .annotationLayer .textWidgetAnnotation textarea[disabled]{background:none;border:1px solid transparent;cursor:not-allowed}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input:hover,.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.radioButton input:hover,.adf-pdf-viewer .annotationLayer .choiceWidgetAnnotation select:hover,.adf-pdf-viewer .annotationLayer .textWidgetAnnotation input:hover,.adf-pdf-viewer .annotationLayer .textWidgetAnnotation textarea:hover{border:1px solid #000}.adf-pdf-viewer .annotationLayer .choiceWidgetAnnotation select:focus,.adf-pdf-viewer .annotationLayer .textWidgetAnnotation input:focus,.adf-pdf-viewer .annotationLayer .textWidgetAnnotation textarea:focus{background:none;border:1px solid transparent}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input:checked:after,.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input:checked:before,.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.radioButton input:checked:before{background-color:#000;content:\"\";display:block;position:absolute}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input:checked:after,.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input:checked:before{height:80%;left:45%;width:1px}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input:checked:before{transform:rotate(45deg)}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input:checked:after{transform:rotate(-45deg)}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.radioButton input:checked:before{border-radius:50%;height:50%;left:30%;top:20%;width:50%}.adf-pdf-viewer .annotationLayer .textWidgetAnnotation input.comb{font-family:monospace;padding-left:2px;padding-right:0}.adf-pdf-viewer .annotationLayer .textWidgetAnnotation input.comb:focus{width:115%}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input,.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.radioButton input{-moz-appearance:none;-webkit-appearance:none;appearance:none;padding:0}.adf-pdf-viewer .annotationLayer .popupWrapper{position:absolute;width:20em}.adf-pdf-viewer .annotationLayer .popup{background-color:#ff9;border-radius:2px;box-shadow:0 2px 5px #888;cursor:pointer;font:message-box;font-size:9px;margin-left:5px;max-width:20em;padding:6px;position:absolute;word-wrap:break-word;z-index:200}.adf-pdf-viewer .annotationLayer .popup>*{font-size:9px}.adf-pdf-viewer .annotationLayer .popup h1{display:inline-block}.adf-pdf-viewer .annotationLayer .popup span{display:inline-block;margin-left:5px}.adf-pdf-viewer .annotationLayer .popup p{border-top:1px solid #333;margin-top:2px;padding-top:2px}.adf-pdf-viewer .annotationLayer .caretAnnotation,.adf-pdf-viewer .annotationLayer .circleAnnotation svg ellipse,.adf-pdf-viewer .annotationLayer .fileAttachmentAnnotation,.adf-pdf-viewer .annotationLayer .freeTextAnnotation,.adf-pdf-viewer .annotationLayer .highlightAnnotation,.adf-pdf-viewer .annotationLayer .inkAnnotation svg polyline,.adf-pdf-viewer .annotationLayer .lineAnnotation svg line,.adf-pdf-viewer .annotationLayer .polygonAnnotation svg polygon,.adf-pdf-viewer .annotationLayer .polylineAnnotation svg polyline,.adf-pdf-viewer .annotationLayer .squareAnnotation svg rect,.adf-pdf-viewer .annotationLayer .squigglyAnnotation,.adf-pdf-viewer .annotationLayer .stampAnnotation,.adf-pdf-viewer .annotationLayer .strikeoutAnnotation,.adf-pdf-viewer .annotationLayer .underlineAnnotation{cursor:pointer}.adf-pdf-viewer .pdfViewer .canvasWrapper{overflow:hidden}.adf-pdf-viewer .pdfViewer .page{background-clip:content-box;background-color:#fff;border:9px solid transparent;direction:ltr;height:1056px;margin:1px auto -8px;overflow:visible;position:relative;width:816px}.adf-pdf-viewer .pdfViewer.removePageBorders .page{border:none;margin:0 auto 10px}.adf-pdf-viewer .pdfViewer.singlePageView{display:inline-block}.adf-pdf-viewer .pdfViewer.singlePageView .page{border:none;margin:0}.adf-pdf-viewer .pdfViewer.scrollHorizontal,.adf-pdf-viewer .pdfViewer.scrollWrapped,.adf-pdf-viewer .spread{margin-left:3.5px;margin-right:3.5px;text-align:center}.adf-pdf-viewer .pdfViewer.scrollHorizontal,.adf-pdf-viewer .spread{white-space:nowrap}.adf-pdf-viewer .pdfViewer.removePageBorders,.adf-pdf-viewer .pdfViewer.scrollHorizontal .spread,.adf-pdf-viewer .pdfViewer.scrollWrapped .spread{margin-left:0;margin-right:0}.adf-pdf-viewer .pdfViewer.scrollHorizontal .page,.adf-pdf-viewer .pdfViewer.scrollHorizontal .spread,.adf-pdf-viewer .pdfViewer.scrollWrapped .page,.adf-pdf-viewer .pdfViewer.scrollWrapped .spread,.adf-pdf-viewer .spread .page{display:inline-block;vertical-align:middle}.adf-pdf-viewer .pdfViewer.scrollHorizontal .page,.adf-pdf-viewer .pdfViewer.scrollWrapped .page,.adf-pdf-viewer .spread .page{margin-left:-3.5px;margin-right:-3.5px}.adf-pdf-viewer .pdfViewer.removePageBorders.scrollHorizontal .page,.adf-pdf-viewer .pdfViewer.removePageBorders.scrollWrapped .page,.adf-pdf-viewer .pdfViewer.removePageBorders .spread .page{margin-left:5px;margin-right:5px}.adf-pdf-viewer .pdfViewer .page canvas{display:block;margin:0}.adf-pdf-viewer .pdfViewer .page canvas[hidden]{display:none}.adf-pdf-viewer .pdfViewer .page .loadingIcon{bottom:0;display:block;left:0;position:absolute;right:0;top:0}.adf-pdf-viewer .pdfPresentationMode .pdfViewer{margin-left:0;margin-right:0}.adf-pdf-viewer .pdfPresentationMode .pdfViewer .page,.adf-pdf-viewer .pdfPresentationMode .pdfViewer .spread{display:block}.adf-pdf-viewer .pdfPresentationMode .pdfViewer .page,.adf-pdf-viewer .pdfPresentationMode .pdfViewer.removePageBorders .page{margin-left:auto;margin-right:auto}.adf-pdf-viewer .pdfPresentationMode:-webkit-full-screen .pdfViewer .page{border:0;margin-bottom:100%}.adf-pdf-viewer .pdfPresentationMode:fullscreen .pdfViewer .page{border:0;margin-bottom:100%}.adf-pdf-viewer .textLayer{border:1px solid grey;bottom:0;left:0;line-height:1;opacity:.2;overflow:hidden;position:absolute;right:0;top:0}.adf-pdf-viewer .textLayer>div{color:transparent;cursor:text;position:absolute;transform-origin:0 0;white-space:pre}.adf-pdf-viewer .textLayer .adf-highlight{background-color:#b400aa;border-radius:4px;margin:-1px;padding:1px}.adf-pdf-viewer .textLayer .adf-highlight.adf-begin{border-radius:4px 0 0 4px}.adf-pdf-viewer .textLayer .adf-highlight.adf-end{border-radius:0 4px 4px 0}.adf-pdf-viewer .textLayer .adf-highlight.adf-middle{border-radius:0}.adf-pdf-viewer .textLayer .adf-highlight.adf-selected{background-color:#006400}.adf-pdf-viewer .textLayer::selection{background:#00f}.adf-pdf-viewer .textLayer::-moz-selection{background:#00f}.adf-pdf-viewer .textLayer .adf-endOfContent{-moz-user-select:none;-ms-user-select:none;-webkit-user-select:none;bottom:0;cursor:default;display:block;left:0;position:absolute;right:0;top:100%;user-select:none;z-index:-1}.adf-pdf-viewer .textLayer .adf-endOfContent.adf-active{top:0}.adf-pdf-viewer .adf-annotationLayer section{position:absolute}.adf-pdf-viewer .adf-annotationLayer .adf-linkAnnotation>a{background:url(\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\") 0 0 repeat;font-size:1em;height:100%;left:0;position:absolute;top:0;width:100%}.adf-pdf-viewer .adf-annotationLayer .adf-linkAnnotation>a:hover{background:#ff0;box-shadow:0 2px 10px #ff0;opacity:.2}.adf-pdf-viewer .adf-annotationLayer .adf-textAnnotation img{cursor:pointer;position:absolute}.adf-pdf-viewer .adf-annotationLayer .adf-popupWrapper{position:absolute;width:20em}.adf-pdf-viewer .adf-annotationLayer .adf-popup{background-color:#ff9;border-radius:2px;box-shadow:0 2px 5px #333;cursor:pointer;margin-left:5px;max-width:20em;padding:.6em;position:absolute;word-wrap:break-word;z-index:200}.adf-pdf-viewer .adf-annotationLayer .adf-popup h1{border-bottom:1px solid #000;font-size:1em;padding-bottom:.2em}.adf-pdf-viewer .adf-annotationLayer .adf-popup p{padding-top:.2em}.adf-pdf-viewer .adf-annotationLayer .adf-fileAttachmentAnnotation,.adf-pdf-viewer .adf-annotationLayer .adf-highlightAnnotation,.adf-pdf-viewer .adf-annotationLayer .adf-squigglyAnnotation,.adf-pdf-viewer .adf-annotationLayer .adf-strikeoutAnnotation,.adf-pdf-viewer .adf-annotationLayer .adf-underlineAnnotation{cursor:pointer}.adf-pdf-viewer .adf-pdfViewer .canvasWrapper{overflow:hidden}.adf-pdf-viewer .adf-pdfViewer .page{background-clip:content-box;background-color:#fff;border:9px solid transparent;direction:ltr;height:1056px;margin:1px auto -8px;overflow:visible;position:relative;width:816px}.adf-pdf-viewer .adf-pdfViewer .page canvas{display:block;margin:0}.adf-pdf-viewer .adf-pdfViewer .page .adf-loadingIcon{bottom:0;display:block;left:0;position:absolute;right:0;top:0}.adf-pdf-viewer .adf-pdfViewer .page *{margin:0;padding:0}.adf-pdf-viewer .adf-pdfViewer.adf-removePageBorders .adf-page{border:none;margin:0 auto 10px}.adf-pdf-viewer .adf-pdfViewer .adf-loadingIcon{animation:load8 1.1s linear infinite;border:1.1em solid rgba(3,0,2,.2);border-left-color:#030002;border-radius:50%;font-size:5px;height:100px;left:50%!important;margin-left:-50px;margin-top:-50px;text-indent:-9999em;top:50%!important;transform:translateZ(0);width:100px}.adf-pdf-viewer .adf-pdfViewer .adf-loadingIcon:after{border-radius:50%}.adf-pdf-viewer .adf-hidden,.adf-pdf-viewer [hidden]{display:none!important}@keyframes load8{0%{transform:rotate(0deg)}to{transform:rotate(1turn)}}.adf-viewer-pdf-viewer{-webkit-overflow-scrolling:touch;bottom:0;left:0;outline:none;overflow:auto;position:absolute;right:0;top:0}html[dir=ltr] .adf-viewer-pdf-viewer{box-shadow:inset 1px 0 0 hsla(0,0%,100%,.05)}html[dir=rtl] .adf-viewer-pdf-viewer{box-shadow:inset -1px 0 0 hsla(0,0%,100%,.05)}"], encapsulation: 2 });
PdfViewerComponent.ctorParameters = () => [
    { type: MatDialog },
    { type: RenderingQueueServices },
    { type: LogService },
    { type: AppConfigService }
];
PdfViewerComponent.propDecorators = {
    urlFile: [{ type: Input }],
    blobFile: [{ type: Input }],
    nameFile: [{ type: Input }],
    showToolbar: [{ type: Input }],
    allowThumbnails: [{ type: Input }],
    thumbnailsTemplate: [{ type: Input }],
    cacheType: [{ type: Input }],
    rendered: [{ type: Output }],
    error: [{ type: Output }],
    close: [{ type: Output }],
    handleKeyboardEvent: [{ type: HostListener, args: ['document:keydown', ['$event'],] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PdfViewerComponent, [{
        type: Component,
        args: [{
                selector: 'adf-pdf-viewer',
                template: "<div class=\"adf-pdf-viewer__container\">\n    <ng-container *ngIf=\"showThumbnails\">\n        <div class=\"adf-pdf-viewer__thumbnails\">\n            <div class=\"adf-thumbnails-template__container\">\n                <div class=\"adf-thumbnails-template__buttons\">\n                    <button\n                        mat-icon-button\n                        data-automation-id='adf-thumbnails-close'\n                        (click)=\"toggleThumbnails()\"\n                        [attr.aria-label]=\"'ADF_VIEWER.ARIA.THUMBNAILS_PANLEL_CLOSE' | translate\"\n                        title=\"{{ 'ADF_VIEWER.ACTIONS.CLOSE' | translate }}\">\n                        <mat-icon>close</mat-icon>\n                    </button>\n                </div>\n\n                <ng-container *ngIf=\"thumbnailsTemplate\">\n                    <ng-container *ngTemplateOutlet=\"thumbnailsTemplate;context:pdfThumbnailsContext\"></ng-container>\n                </ng-container>\n\n                <ng-container *ngIf=\"!thumbnailsTemplate\">\n                    <adf-pdf-thumbnails (close)=\"toggleThumbnails()\" [pdfViewer]=\"pdfViewer\"></adf-pdf-thumbnails>\n                </ng-container>\n            </div>\n        </div>\n    </ng-container>\n\n    <div class=\"adf-pdf-viewer__content\">\n        <div [id]=\"randomPdfId+'-viewer-pdf-viewer'\" class=\"adf-viewer-pdf-viewer\" (window:resize)=\"onResize()\">\n            <div [id]=\"randomPdfId+'-viewer-viewerPdf'\" class=\"adf-pdfViewer\" role=\"document\" tabindex=\"0\" aria-expanded=\"true\">\n                <div id=\"loader-container\" class=\"adf-loader-container\">\n                    <div class=\"adf-loader-item\">\n                        <mat-progress-bar mode=\"indeterminate\"></mat-progress-bar>\n                    </div>\n                </div>\n            </div>\n        </div>\n    </div>\n</div>\n\n<div class=\"adf-pdf-viewer__toolbar\" *ngIf=\"showToolbar\">\n    <adf-toolbar>\n\n        <ng-container *ngIf=\"allowThumbnails\">\n            <button mat-icon-button\n                    [attr.aria-label]=\"'ADF_VIEWER.ARIA.THUMBNAILS' | translate\"\n                    [attr.aria-expanded]=\"showThumbnails\"\n                    data-automation-id=\"adf-thumbnails-button\"\n                    [disabled]=\"isPanelDisabled\"\n                    (click)=\"toggleThumbnails()\">\n                <mat-icon>dashboard</mat-icon>\n            </button>\n            <adf-toolbar-divider></adf-toolbar-divider>\n        </ng-container>\n\n        <button\n            id=\"viewer-previous-page-button\"\n            title=\"{{ 'ADF_VIEWER.ARIA.PREVIOUS_PAGE' | translate }}\"\n            attr.aria-label=\"{{ 'ADF_VIEWER.ARIA.PREVIOUS_PAGE' | translate }}\"\n            mat-icon-button\n            (click)=\"previousPage()\">\n            <mat-icon>keyboard_arrow_up</mat-icon>\n        </button>\n\n        <button\n            id=\"viewer-next-page-button\"\n            title=\"{{ 'ADF_VIEWER.ARIA.NEXT_PAGE' | translate }}\"\n            attr.aria-label=\"{{ 'ADF_VIEWER.ARIA.NEXT_PAGE' | translate }}\"\n            mat-icon-button\n            (click)=\"nextPage()\">\n            <mat-icon>keyboard_arrow_down</mat-icon>\n        </button>\n\n        <div class=\"adf-pdf-viewer__toolbar-page-selector\">\n            <span>{{ 'ADF_VIEWER.PAGE_LABEL.SHOWING' | translate }}</span>\n            <input #page\n                   type=\"text\"\n                   data-automation-id=\"adf-page-selector\"\n                   pattern=\"-?[0-9]*(\\.[0-9]+)?\"\n                   value=\"{{ displayPage }}\"\n                   (keyup.enter)=\"inputPage(page.value)\">\n            <span>{{ 'ADF_VIEWER.PAGE_LABEL.OF' | translate }} {{ totalPages }}</span>\n        </div>\n\n        <div class=\"adf-viewer__toolbar-page-scale\" data-automation-id=\"adf-page-scale\">\n            {{ currentScaleText }}\n        </div>\n\n        <button\n            id=\"viewer-zoom-in-button\"\n            title=\"{{ 'ADF_VIEWER.ARIA.ZOOM_IN' | translate }}\"\n            attr.aria-label=\"{{ 'ADF_VIEWER.ARIA.ZOOM_IN' | translate }}\"\n            mat-icon-button\n            (click)=\"zoomIn()\">\n            <mat-icon>zoom_in</mat-icon>\n        </button>\n\n        <button\n            id=\"viewer-zoom-out-button\"\n            title=\"{{ 'ADF_VIEWER.ARIA.ZOOM_OUT' | translate }}\"\n            attr.aria-label=\"{{ 'ADF_VIEWER.ARIA.ZOOM_OUT' | translate }}\"\n            mat-icon-button\n            (click)=\"zoomOut()\">\n            <mat-icon>zoom_out</mat-icon>\n        </button>\n\n        <button\n            id=\"viewer-scale-page-button\"\n            title=\"{{ 'ADF_VIEWER.ARIA.FIT_PAGE' | translate }}\"\n            attr.aria-label=\"{{ 'ADF_VIEWER.ARIA.FIT_PAGE' | translate }}\"\n            mat-icon-button\n            (click)=\"pageFit()\">\n            <mat-icon>zoom_out_map</mat-icon>\n        </button>\n\n    </adf-toolbar>\n</div>\n",
                providers: [RenderingQueueServices],
                host: { 'class': 'adf-pdf-viewer' },
                encapsulation: ViewEncapsulation.None,
                styles: [".adf-pdf-viewer .textLayer>span{color:transparent;cursor:text;position:absolute;transform-origin:0 0;white-space:pre}.adf-pdf-viewer .textLayer .highlight{background-color:#b400aa;border-radius:4px;margin:-1px;padding:1px}.adf-pdf-viewer .textLayer .highlight.begin{border-radius:4px 0 0 4px}.adf-pdf-viewer .textLayer .highlight.end{border-radius:0 4px 4px 0}.adf-pdf-viewer .textLayer .highlight.middle{border-radius:0}.adf-pdf-viewer .textLayer .highlight.selected{background-color:#006400}.adf-pdf-viewer .textLayer ::selection{background:#00f}.adf-pdf-viewer .textLayer .endOfContent{-webkit-user-select:none;bottom:0;cursor:default;display:block;left:0;position:absolute;right:0;top:100%;user-select:none;z-index:-1}.adf-pdf-viewer .textLayer .endOfContent.active{top:0}.adf-pdf-viewer .annotationLayer section{position:absolute}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.pushButton>a,.adf-pdf-viewer .annotationLayer .linkAnnotation>a{font-size:1em;height:100%;left:0;position:absolute;top:0;width:100%}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.pushButton>a:hover,.adf-pdf-viewer .annotationLayer .linkAnnotation>a:hover{background:#ff0;box-shadow:0 2px 10px #ff0;opacity:.2}.adf-pdf-viewer .annotationLayer .textAnnotation img{cursor:pointer;position:absolute}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input,.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.radioButton input,.adf-pdf-viewer .annotationLayer .choiceWidgetAnnotation select,.adf-pdf-viewer .annotationLayer .textWidgetAnnotation input,.adf-pdf-viewer .annotationLayer .textWidgetAnnotation textarea{background-color:rgba(0,54,255,.13);border:1px solid transparent;box-sizing:border-box;font-size:9px;height:100%;margin:0;padding:0 3px;vertical-align:top;width:100%}.adf-pdf-viewer .annotationLayer .choiceWidgetAnnotation select option{padding:0}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.radioButton input{border-radius:50%}.adf-pdf-viewer .annotationLayer .textWidgetAnnotation textarea{font:message-box;font-size:9px;resize:none}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input[disabled],.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.radioButton input[disabled],.adf-pdf-viewer .annotationLayer .choiceWidgetAnnotation select[disabled],.adf-pdf-viewer .annotationLayer .textWidgetAnnotation input[disabled],.adf-pdf-viewer .annotationLayer .textWidgetAnnotation textarea[disabled]{background:none;border:1px solid transparent;cursor:not-allowed}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input:hover,.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.radioButton input:hover,.adf-pdf-viewer .annotationLayer .choiceWidgetAnnotation select:hover,.adf-pdf-viewer .annotationLayer .textWidgetAnnotation input:hover,.adf-pdf-viewer .annotationLayer .textWidgetAnnotation textarea:hover{border:1px solid #000}.adf-pdf-viewer .annotationLayer .choiceWidgetAnnotation select:focus,.adf-pdf-viewer .annotationLayer .textWidgetAnnotation input:focus,.adf-pdf-viewer .annotationLayer .textWidgetAnnotation textarea:focus{background:none;border:1px solid transparent}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input:checked:after,.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input:checked:before,.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.radioButton input:checked:before{background-color:#000;content:\"\";display:block;position:absolute}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input:checked:after,.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input:checked:before{height:80%;left:45%;width:1px}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input:checked:before{transform:rotate(45deg)}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input:checked:after{transform:rotate(-45deg)}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.radioButton input:checked:before{border-radius:50%;height:50%;left:30%;top:20%;width:50%}.adf-pdf-viewer .annotationLayer .textWidgetAnnotation input.comb{font-family:monospace;padding-left:2px;padding-right:0}.adf-pdf-viewer .annotationLayer .textWidgetAnnotation input.comb:focus{width:115%}.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.checkBox input,.adf-pdf-viewer .annotationLayer .buttonWidgetAnnotation.radioButton input{-moz-appearance:none;-webkit-appearance:none;appearance:none;padding:0}.adf-pdf-viewer .annotationLayer .popupWrapper{position:absolute;width:20em}.adf-pdf-viewer .annotationLayer .popup{background-color:#ff9;border-radius:2px;box-shadow:0 2px 5px #888;cursor:pointer;font:message-box;font-size:9px;margin-left:5px;max-width:20em;padding:6px;position:absolute;word-wrap:break-word;z-index:200}.adf-pdf-viewer .annotationLayer .popup>*{font-size:9px}.adf-pdf-viewer .annotationLayer .popup h1{display:inline-block}.adf-pdf-viewer .annotationLayer .popup span{display:inline-block;margin-left:5px}.adf-pdf-viewer .annotationLayer .popup p{border-top:1px solid #333;margin-top:2px;padding-top:2px}.adf-pdf-viewer .annotationLayer .caretAnnotation,.adf-pdf-viewer .annotationLayer .circleAnnotation svg ellipse,.adf-pdf-viewer .annotationLayer .fileAttachmentAnnotation,.adf-pdf-viewer .annotationLayer .freeTextAnnotation,.adf-pdf-viewer .annotationLayer .highlightAnnotation,.adf-pdf-viewer .annotationLayer .inkAnnotation svg polyline,.adf-pdf-viewer .annotationLayer .lineAnnotation svg line,.adf-pdf-viewer .annotationLayer .polygonAnnotation svg polygon,.adf-pdf-viewer .annotationLayer .polylineAnnotation svg polyline,.adf-pdf-viewer .annotationLayer .squareAnnotation svg rect,.adf-pdf-viewer .annotationLayer .squigglyAnnotation,.adf-pdf-viewer .annotationLayer .stampAnnotation,.adf-pdf-viewer .annotationLayer .strikeoutAnnotation,.adf-pdf-viewer .annotationLayer .underlineAnnotation{cursor:pointer}.adf-pdf-viewer .pdfViewer .canvasWrapper{overflow:hidden}.adf-pdf-viewer .pdfViewer .page{background-clip:content-box;background-color:#fff;border:9px solid transparent;direction:ltr;height:1056px;margin:1px auto -8px;overflow:visible;position:relative;width:816px}.adf-pdf-viewer .pdfViewer.removePageBorders .page{border:none;margin:0 auto 10px}.adf-pdf-viewer .pdfViewer.singlePageView{display:inline-block}.adf-pdf-viewer .pdfViewer.singlePageView .page{border:none;margin:0}.adf-pdf-viewer .pdfViewer.scrollHorizontal,.adf-pdf-viewer .pdfViewer.scrollWrapped,.adf-pdf-viewer .spread{margin-left:3.5px;margin-right:3.5px;text-align:center}.adf-pdf-viewer .pdfViewer.scrollHorizontal,.adf-pdf-viewer .spread{white-space:nowrap}.adf-pdf-viewer .pdfViewer.removePageBorders,.adf-pdf-viewer .pdfViewer.scrollHorizontal .spread,.adf-pdf-viewer .pdfViewer.scrollWrapped .spread{margin-left:0;margin-right:0}.adf-pdf-viewer .pdfViewer.scrollHorizontal .page,.adf-pdf-viewer .pdfViewer.scrollHorizontal .spread,.adf-pdf-viewer .pdfViewer.scrollWrapped .page,.adf-pdf-viewer .pdfViewer.scrollWrapped .spread,.adf-pdf-viewer .spread .page{display:inline-block;vertical-align:middle}.adf-pdf-viewer .pdfViewer.scrollHorizontal .page,.adf-pdf-viewer .pdfViewer.scrollWrapped .page,.adf-pdf-viewer .spread .page{margin-left:-3.5px;margin-right:-3.5px}.adf-pdf-viewer .pdfViewer.removePageBorders.scrollHorizontal .page,.adf-pdf-viewer .pdfViewer.removePageBorders.scrollWrapped .page,.adf-pdf-viewer .pdfViewer.removePageBorders .spread .page{margin-left:5px;margin-right:5px}.adf-pdf-viewer .pdfViewer .page canvas{display:block;margin:0}.adf-pdf-viewer .pdfViewer .page canvas[hidden]{display:none}.adf-pdf-viewer .pdfViewer .page .loadingIcon{bottom:0;display:block;left:0;position:absolute;right:0;top:0}.adf-pdf-viewer .pdfPresentationMode .pdfViewer{margin-left:0;margin-right:0}.adf-pdf-viewer .pdfPresentationMode .pdfViewer .page,.adf-pdf-viewer .pdfPresentationMode .pdfViewer .spread{display:block}.adf-pdf-viewer .pdfPresentationMode .pdfViewer .page,.adf-pdf-viewer .pdfPresentationMode .pdfViewer.removePageBorders .page{margin-left:auto;margin-right:auto}.adf-pdf-viewer .pdfPresentationMode:-webkit-full-screen .pdfViewer .page{border:0;margin-bottom:100%}.adf-pdf-viewer .pdfPresentationMode:fullscreen .pdfViewer .page{border:0;margin-bottom:100%}.adf-pdf-viewer .textLayer{border:1px solid grey;bottom:0;left:0;line-height:1;opacity:.2;overflow:hidden;position:absolute;right:0;top:0}.adf-pdf-viewer .textLayer>div{color:transparent;cursor:text;position:absolute;transform-origin:0 0;white-space:pre}.adf-pdf-viewer .textLayer .adf-highlight{background-color:#b400aa;border-radius:4px;margin:-1px;padding:1px}.adf-pdf-viewer .textLayer .adf-highlight.adf-begin{border-radius:4px 0 0 4px}.adf-pdf-viewer .textLayer .adf-highlight.adf-end{border-radius:0 4px 4px 0}.adf-pdf-viewer .textLayer .adf-highlight.adf-middle{border-radius:0}.adf-pdf-viewer .textLayer .adf-highlight.adf-selected{background-color:#006400}.adf-pdf-viewer .textLayer::selection{background:#00f}.adf-pdf-viewer .textLayer::-moz-selection{background:#00f}.adf-pdf-viewer .textLayer .adf-endOfContent{-moz-user-select:none;-ms-user-select:none;-webkit-user-select:none;bottom:0;cursor:default;display:block;left:0;position:absolute;right:0;top:100%;user-select:none;z-index:-1}.adf-pdf-viewer .textLayer .adf-endOfContent.adf-active{top:0}.adf-pdf-viewer .adf-annotationLayer section{position:absolute}.adf-pdf-viewer .adf-annotationLayer .adf-linkAnnotation>a{background:url(\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\") 0 0 repeat;font-size:1em;height:100%;left:0;position:absolute;top:0;width:100%}.adf-pdf-viewer .adf-annotationLayer .adf-linkAnnotation>a:hover{background:#ff0;box-shadow:0 2px 10px #ff0;opacity:.2}.adf-pdf-viewer .adf-annotationLayer .adf-textAnnotation img{cursor:pointer;position:absolute}.adf-pdf-viewer .adf-annotationLayer .adf-popupWrapper{position:absolute;width:20em}.adf-pdf-viewer .adf-annotationLayer .adf-popup{background-color:#ff9;border-radius:2px;box-shadow:0 2px 5px #333;cursor:pointer;margin-left:5px;max-width:20em;padding:.6em;position:absolute;word-wrap:break-word;z-index:200}.adf-pdf-viewer .adf-annotationLayer .adf-popup h1{border-bottom:1px solid #000;font-size:1em;padding-bottom:.2em}.adf-pdf-viewer .adf-annotationLayer .adf-popup p{padding-top:.2em}.adf-pdf-viewer .adf-annotationLayer .adf-fileAttachmentAnnotation,.adf-pdf-viewer .adf-annotationLayer .adf-highlightAnnotation,.adf-pdf-viewer .adf-annotationLayer .adf-squigglyAnnotation,.adf-pdf-viewer .adf-annotationLayer .adf-strikeoutAnnotation,.adf-pdf-viewer .adf-annotationLayer .adf-underlineAnnotation{cursor:pointer}.adf-pdf-viewer .adf-pdfViewer .canvasWrapper{overflow:hidden}.adf-pdf-viewer .adf-pdfViewer .page{background-clip:content-box;background-color:#fff;border:9px solid transparent;direction:ltr;height:1056px;margin:1px auto -8px;overflow:visible;position:relative;width:816px}.adf-pdf-viewer .adf-pdfViewer .page canvas{display:block;margin:0}.adf-pdf-viewer .adf-pdfViewer .page .adf-loadingIcon{bottom:0;display:block;left:0;position:absolute;right:0;top:0}.adf-pdf-viewer .adf-pdfViewer .page *{margin:0;padding:0}.adf-pdf-viewer .adf-pdfViewer.adf-removePageBorders .adf-page{border:none;margin:0 auto 10px}.adf-pdf-viewer .adf-pdfViewer .adf-loadingIcon{animation:load8 1.1s linear infinite;border:1.1em solid rgba(3,0,2,.2);border-left-color:#030002;border-radius:50%;font-size:5px;height:100px;left:50%!important;margin-left:-50px;margin-top:-50px;text-indent:-9999em;top:50%!important;transform:translateZ(0);width:100px}.adf-pdf-viewer .adf-pdfViewer .adf-loadingIcon:after{border-radius:50%}.adf-pdf-viewer .adf-hidden,.adf-pdf-viewer [hidden]{display:none!important}@keyframes load8{0%{transform:rotate(0deg)}to{transform:rotate(1turn)}}.adf-viewer-pdf-viewer{-webkit-overflow-scrolling:touch;bottom:0;left:0;outline:none;overflow:auto;position:absolute;right:0;top:0}html[dir=ltr] .adf-viewer-pdf-viewer{box-shadow:inset 1px 0 0 hsla(0,0%,100%,.05)}html[dir=rtl] .adf-viewer-pdf-viewer{box-shadow:inset -1px 0 0 hsla(0,0%,100%,.05)}"]
            }]
    }], function () { return [{ type: ɵngcc1.MatDialog }, { type: ɵngcc2.RenderingQueueServices }, { type: ɵngcc3.LogService }, { type: ɵngcc4.AppConfigService }]; }, { showToolbar: [{
            type: Input
        }], allowThumbnails: [{
            type: Input
        }], thumbnailsTemplate: [{
            type: Input
        }], cacheType: [{
            type: Input
        }], rendered: [{
            type: Output
        }], error: [{
            type: Output
        }], close: [{
            type: Output
        }], handleKeyboardEvent: [{
            type: HostListener,
            args: ['document:keydown', ['$event']]
        }], urlFile: [{
            type: Input
        }], blobFile: [{
            type: Input
        }], nameFile: [{
            type: Input
        }] }); })();
