// @alfresco/adf-core/esm2015/services/download-zip.service.js
import { Injectable } from '@angular/core';
import { from, throwError } from 'rxjs';
import { LogService } from './log.service';
import { AlfrescoApiService } from './alfresco-api.service';
import { catchError } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "./alfresco-api.service";
import * as i2 from "./log.service";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './alfresco-api.service';
import * as ɵngcc2 from './log.service';
export class DownloadZipService {
    constructor(apiService, logService) {
        this.apiService = apiService;
        this.logService = logService;
    }
    createDownload(payload) {
        return from(this.apiService.getInstance().core.downloadsApi.createDownload(payload)).pipe(catchError((err) => this.handleError(err)));
    }
    getContentUrl(nodeId, attachment) {
        return this.apiService.getInstance().content.getContentUrl(nodeId, attachment);
    }
    getNode(nodeId) {
        return from(this.apiService.getInstance().core.nodesApi.getNode(nodeId));
    }
    getDownload(downloadId) {
        return from(this.apiService.getInstance().core.downloadsApi.getDownload(downloadId));
    }
    cancelDownload(downloadId) {
        this.apiService.getInstance().core.downloadsApi.cancelDownload(downloadId);
    }
    handleError(error) {
        this.logService.error(error);
        return throwError(error || 'Server error');
    }
}
DownloadZipService.ɵfac = function DownloadZipService_Factory(t) { return new (t || DownloadZipService)(ɵngcc0.ɵɵinject(ɵngcc1.AlfrescoApiService), ɵngcc0.ɵɵinject(ɵngcc2.LogService)); };
DownloadZipService.ɵprov = i0.ɵɵdefineInjectable({ factory: function DownloadZipService_Factory() { return new DownloadZipService(i0.ɵɵinject(i1.AlfrescoApiService), i0.ɵɵinject(i2.LogService)); }, token: DownloadZipService, providedIn: "root" });
DownloadZipService.ctorParameters = () => [
    { type: AlfrescoApiService },
    { type: LogService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DownloadZipService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: ɵngcc1.AlfrescoApiService }, { type: ɵngcc2.LogService }]; }, null); })();
