// @alfresco/adf-content-services/esm2015/lib/search/components/search-form/search-form.component.js
/*!
 * @license
 * Copyright 2019 Alfresco Software, Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Component, EventEmitter, Inject, Output, ViewEncapsulation } from '@angular/core';
import { SearchQueryBuilderService } from '../../services/search-query-builder.service';
import { SEARCH_QUERY_SERVICE_TOKEN } from '../../search-query-service.token';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '@angular/material/button';
import * as ɵngcc3 from '@angular/material/menu';
import * as ɵngcc4 from '@angular/material/icon';
import * as ɵngcc5 from '@ngx-translate/core';
import * as ɵngcc6 from '../../services/search-query-builder.service';

function SearchFormComponent_ng_container_0_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "button", 1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementStart(3, "span", 2);
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵpipe(5, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const forms_r1 = ɵngcc0.ɵɵnextContext().ngIf;
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("title", ɵngcc0.ɵɵpipeBind1(2, 2, ctx_r2.getSelected(forms_r1)));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(5, 4, ctx_r2.getSelected(forms_r1)), " ");
} }
function SearchFormComponent_ng_container_0_ng_container_2_button_11_Template(rf, ctx) { if (rf & 1) {
    const _r10 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 9);
    ɵngcc0.ɵɵlistener("click", function SearchFormComponent_ng_container_0_ng_container_2_button_11_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r10); const form_r8 = ctx.$implicit; const ctx_r9 = ɵngcc0.ɵɵnextContext(3); return ctx_r9.onSelectionChange(form_r8); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "translate");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const form_r8 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 1, form_r8.name), " ");
} }
function SearchFormComponent_ng_container_0_ng_container_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "button", 3, 4);
    ɵngcc0.ɵɵpipe(3, "translate");
    ɵngcc0.ɵɵelementStart(4, "span", 2);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵpipe(6, "translate");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(7, "mat-icon", 5);
    ɵngcc0.ɵɵtext(8, "expand_more");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(9, "mat-menu", 6, 7);
    ɵngcc0.ɵɵtemplate(11, SearchFormComponent_ng_container_0_ng_container_2_button_11_Template, 3, 3, "button", 8);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const _r5 = ɵngcc0.ɵɵreference(2);
    const _r6 = ɵngcc0.ɵɵreference(10);
    const forms_r1 = ɵngcc0.ɵɵnextContext().ngIf;
    const ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("matMenuTriggerFor", _r6)("title", ɵngcc0.ɵɵpipeBind1(3, 7, ctx_r3.getSelected(forms_r1)))("matMenuTriggerRestoreFocus", true);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(6, 9, ctx_r3.getSelected(forms_r1)), " ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵclassProp("adf-search-form-icon-selected", _r5.menuOpen);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵproperty("ngForOf", forms_r1);
} }
function SearchFormComponent_ng_container_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, SearchFormComponent_ng_container_0_ng_container_1_Template, 6, 6, "ng-container", 0);
    ɵngcc0.ɵɵtemplate(2, SearchFormComponent_ng_container_0_ng_container_2_Template, 12, 11, "ng-container", 0);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const forms_r1 = ctx.ngIf;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", forms_r1.length === 1);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", forms_r1.length > 1);
} }
export class SearchFormComponent {
    constructor(queryBuilder) {
        this.queryBuilder = queryBuilder;
        this.formChange = new EventEmitter();
    }
    onSelectionChange(form) {
        this.queryBuilder.updateSelectedConfiguration(form.index);
        this.formChange.emit(form);
    }
    getSelected(forms) {
        var _a;
        return (_a = forms.find((form) => form.selected)) === null || _a === void 0 ? void 0 : _a.name;
    }
}
SearchFormComponent.ɵfac = function SearchFormComponent_Factory(t) { return new (t || SearchFormComponent)(ɵngcc0.ɵɵdirectiveInject(SEARCH_QUERY_SERVICE_TOKEN)); };
SearchFormComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SearchFormComponent, selectors: [["adf-search-form"]], outputs: { formChange: "formChange" }, decls: 2, vars: 3, consts: [[4, "ngIf"], ["disableRipple", "", "mat-button", "", 1, "adf-search-form", 3, "title"], [1, "adf-search-form-title"], ["disableRipple", "", "mat-button", "", 1, "adf-search-form", 3, "matMenuTriggerFor", "title", "matMenuTriggerRestoreFocus"], ["menuTrigger", "matMenuTrigger"], [1, "adf-search-form-icon"], ["backdropClass", "adf-search-form-menu"], ["menu", "matMenu"], ["mat-menu-item", "", 3, "click", 4, "ngFor", "ngForOf"], ["mat-menu-item", "", 3, "click"]], template: function SearchFormComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, SearchFormComponent_ng_container_0_Template, 3, 2, "ng-container", 0);
        ɵngcc0.ɵɵpipe(1, "async");
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ɵngcc0.ɵɵpipeBind1(1, 1, ctx.queryBuilder.searchForms));
    } }, directives: [ɵngcc1.NgIf, ɵngcc2.MatButton, ɵngcc3.MatMenuTrigger, ɵngcc4.MatIcon, ɵngcc3._MatMenu, ɵngcc1.NgForOf, ɵngcc3.MatMenuItem], pipes: [ɵngcc1.AsyncPipe, ɵngcc5.TranslatePipe], styles: [""], encapsulation: 2 });
SearchFormComponent.ctorParameters = () => [
    { type: SearchQueryBuilderService, decorators: [{ type: Inject, args: [SEARCH_QUERY_SERVICE_TOKEN,] }] }
];
SearchFormComponent.propDecorators = {
    formChange: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SearchFormComponent, [{
        type: Component,
        args: [{
                selector: 'adf-search-form',
                template: "<ng-container *ngIf=\"queryBuilder.searchForms | async as forms\">\n\n    <ng-container *ngIf=\"forms.length === 1\">\n        <button class=\"adf-search-form\" disableRipple mat-button [title]=\"getSelected(forms) | translate\">\n            <span class=\"adf-search-form-title\">\n                    {{ getSelected(forms) | translate }}\n            </span>\n        </button>\n    </ng-container>\n\n    <ng-container  *ngIf=\"forms.length > 1\">\n        <button class=\"adf-search-form\"\n                [matMenuTriggerFor]=\"menu\"\n                #menuTrigger=\"matMenuTrigger\"\n                disableRipple\n                mat-button\n                [title]=\"getSelected(forms) | translate\"\n                [matMenuTriggerRestoreFocus]=\"true\">\n\n            <span class=\"adf-search-form-title\" >\n                    {{ getSelected(forms) | translate }}\n            </span>\n\n            <mat-icon [class.adf-search-form-icon-selected]=\"menuTrigger.menuOpen\" class=\"adf-search-form-icon\">expand_more</mat-icon>\n        </button>\n\n        <mat-menu #menu=\"matMenu\" backdropClass=\"adf-search-form-menu\">\n            <button mat-menu-item *ngFor=\"let form of forms\" (click)=\"onSelectionChange(form)\">\n                {{form.name | translate}}\n            </button>\n        </mat-menu>\n    </ng-container>\n</ng-container>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [""]
            }]
    }], function () { return [{ type: ɵngcc6.SearchQueryBuilderService, decorators: [{
                type: Inject,
                args: [SEARCH_QUERY_SERVICE_TOKEN]
            }] }]; }, { formChange: [{
            type: Output
        }] }); })();
